// common-ui/lib/components/NextDialog/NextDialog.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _Modal = _interopRequireDefault(require("@material-ui/core/Modal"));
var _core = require("@material-ui/core");
var _styles = require("@material-ui/core/styles");
var _classnames = _interopRequireDefault(require("classnames"));
var _commonIcons = require("common-icons");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
var _focusTrap = _interopRequireDefault(require("./focusTrap"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root;
  return {
    background: {
      display: function display(props) {
        return props.open ? 'block' : 'none';
      },
      backgroundColor: 'rgba(226, 236, 249, 0.95)',
      width: '100%',
      height: '100%',
      top: 0,
      right: 0,
      position: 'fixed',
      zIndex: 10
    },
    root: (_root = {
      width: function width(props) {
        return props.width ? props.width : 630;
      },
      margin: function margin(props) {
        return props.isCenter ? '50px 0' : '100px auto';
      },
      padding: theme.spacing(2),
      position: 'relative',
      boxShadow: '0 0 18px 0 rgba(0, 62, 137, 0.15)',
      outline: 'none'
    }, (0, _defineProperty2.default)(_root, theme.breakpoints.down('md'), {
      width: function width(props) {
        return props.mdWidth || '85%';
      },
      margin: '30px auto'
    }), (0, _defineProperty2.default)(_root, theme.breakpoints.down('sm'), {
      width: function width(props) {
        return props.smWidth || props.mdWidth || '85%';
      }
    }), _root),
    title: {
      fontWeight: 400,
      marginBottom: theme.spacing(2),
      textAlign: 'center'
    },
    dialogIcon: {
      position: 'relative',
      width: 90,
      margin: '0 auto',
      top: '-70px',
      marginBottom: function marginBottom(props) {
        return props.iconMarginBottom ? props.iconMarginBottom : '-80px';
      },
      fontSize: 80,
      textAlign: 'center',
      color: '#2d76cb'
    },
    closeIconButton: {
      position: 'absolute',
      top: theme.spacing(1),
      left: theme.spacing(1)
    },
    iconButton: {
      '&:focus': {
        outline: '1px solid #666'
      }
    },
    scrollable: {
      overflowY: 'auto',
      display: 'flex',
      flexDirection: 'row',
      alignItems: function alignItems(props) {
        return props.isCenter ? 'center' : 'flex-start';
      },
      justifyContent: 'center',
      flexWrap: 'wrap'
    }
  };
});
var NextDialog = function NextDialog(props) {
  var _props$dataHrlBo;
  var open = props.open,
    onClose = props.onClose;
  var _useState = (0, _react.useState)(false),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    touched = _useState2[0],
    setTouched = _useState2[1];
  var closeButtonRef = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var classes = useStyles(props);
  var hrlBo = (_props$dataHrlBo = props['data-hrl-bo']) !== null && _props$dataHrlBo !== void 0 ? _props$dataHrlBo : 'next-dialog';
  var focusManager = function focusManager(e) {
    var _e$target;
    if (!touched) {
      e.target.removeAttribute('tabIndex');
    }
    if (!((_e$target = e.target) !== null && _e$target !== void 0 && _e$target.id)) {
      e.target.setAttribute('id', Math.random().toString().slice(10));
      (0, _focusTrap.default)(e.target);
    }
    !touched && setTouched(true);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.background,
    onClick: !props.disableBackgroundClick && props.onClose,
    "data-hrl-bo": hrlBo
  }, /*#__PURE__*/_react.default.createElement(_Modal.default, {
    open: open,
    onFocus: focusManager,
    onClose: onClose,
    hideBackdrop: true,
    "aria-labelledby": "simple-modal-title",
    "aria-describedby": "simple-modal-description",
    role: "dialog",
    className: (0, _classnames.default)('modal-dialog', classes.scrollable)
  }, /*#__PURE__*/_react.default.createElement(_core.Paper, {
    variant: "outlined",
    className: classes.root,
    onClick: function onClick(e) {
      return e.stopPropagation();
    }
  }, /*#__PURE__*/_react.default.createElement(NextDialogHeader, (0, _extends2.default)({}, props, {
    hrlBo: hrlBo,
    ref: closeButtonRef
  })), /*#__PURE__*/_react.default.createElement("div", {
    "data-hrl-bo": "".concat(hrlBo, "_content")
  }, props.children))));
};
var NextDialogHeader = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var onClose = props.onClose,
    iconClose = props.iconClose,
    hrlBo = props.hrlBo,
    icon = props.icon,
    title = props.title;
  var classes = useStyles(props);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, icon && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.dialogIcon,
    "data-hrl-bo": "".concat(hrlBo, "_icon")
  }, icon), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.closeIconButton,
    "data-hrl-bo": "".concat(hrlBo, "_close_button")
  }, /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    size: "small",
    color: "primary",
    onClick: onClose,
    ref: ref,
    "aria-label": 'סגור חלון',
    className: classes.iconButton
  }, iconClose ? iconClose : /*#__PURE__*/_react.default.createElement(_commonIcons.IconX, null))), title && /*#__PURE__*/_react.default.createElement(_core.Typography, {
    variant: "h1",
    className: classes.title,
    "data-hrl-bo": "".concat(hrlBo, "_title")
  }, title));
});
NextDialog.propTypes = {
  title: _propTypes.default.string,
  onClose: _propTypes.default.func,
  icon: _propTypes.default.node,
  open: _propTypes.default.bool,
  isCenter: _propTypes.default.bool,
  disableBackgroundClick: _propTypes.default.bool,
  iconClose: _propTypes.default.node,
  width: _propTypes.default.oneOf([_propTypes.default.number, _propTypes.default.string]),
  // support str and num: ['50%', 450, '-10px']
  mdWidth: _propTypes.default.oneOf([_propTypes.default.number, _propTypes.default.string]),
  smWidth: _propTypes.default.oneOf([_propTypes.default.number, _propTypes.default.string]),
  iconMarginBottom: _propTypes.default.string,
  children: _propTypes.default.node,
  'data-hrl-bo': _propTypes.default.string,
  position: _propTypes.default.oneOf(['middle', 'top'])
};
NextDialog.defaultProps = {
  position: 'middle'
};
var _default = NextDialog;
exports.default = _default;
