// common-ui/lib/components/Identification/Identification.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Grid = _interopRequireDefault(require("../Grid"));
var _core = require("@material-ui/core");
var _ = require("../..");
var _useReduxField = _interopRequireWildcard(require("../HOC/useReduxField"));
var _misc = require("../../utils/misc");
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
var _excluded = ["onChange", "value", "grid", "spacing", "useRedux", "name", "required", "helperText", "error", "errors", "proxy"];
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var IdentificationInner = function IdentificationInner(_ref) {
  var onChange = _ref.onChange,
    value = _ref.value,
    grid = _ref.grid,
    spacing = _ref.spacing,
    useRedux = _ref.useRedux,
    name = _ref.name,
    required = _ref.required,
    helperText = _ref.helperText,
    error = _ref.error,
    errors = _ref.errors,
    proxy = _ref.proxy,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var form = useRedux({
    firstName: value.firstName || '',
    lastName: value.lastName || '',
    id: value.id || '',
    birthDate: value.birthDate || ''
  });
  var getFieldProps = form.getFieldProps,
    handleChange = form.handleChange,
    handleBlur = form.handleBlur,
    values = form.values;
  var firstName = values.firstName,
    lastName = values.lastName,
    id = values.id,
    birthDate = values.birthDate;
  var fieldName = name;
  return /*#__PURE__*/_react.default.createElement(_Grid.default, {
    container: true,
    spacing: spacing
  }, /*#__PURE__*/_react.default.createElement(_Grid.default, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.firstName), /*#__PURE__*/_react.default.createElement(_core.TextField, (0, _extends2.default)({
    value: firstName,
    label: "\u05E9\u05DD \u05E4\u05E8\u05D8\u05D9"
  }, getFieldProps('firstName')))), /*#__PURE__*/_react.default.createElement(_Grid.default, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.lastName), /*#__PURE__*/_react.default.createElement(_core.TextField, (0, _extends2.default)({
    value: lastName,
    label: "\u05E9\u05DD \u05DE\u05E9\u05E4\u05D7\u05D4"
  }, getFieldProps('lastName')))), /*#__PURE__*/_react.default.createElement(_Grid.default, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.id), /*#__PURE__*/_react.default.createElement(_.IDInput, (0, _extends2.default)({
    value: id,
    label: "\u05EA\u05E2\u05D5\u05D3\u05EA \u05D6\u05D4\u05D5\u05EA"
  }, getFieldProps('id')))), /*#__PURE__*/_react.default.createElement(_Grid.default, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.birthDate), /*#__PURE__*/_react.default.createElement(_.DatePicker, (0, _extends2.default)({
    value: birthDate,
    autoOk: false,
    onChange: function onChange(value) {
      return handleChange('birthDate')((0, _misc.getEventData)(fieldName, value));
    },
    label: "\u05D9\u05D5\u05DD \u05D4\u05D5\u05DC\u05D3\u05EA"
  }, getFieldProps('birthDate')))));
};
var sizesStruct = _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]);
var singleObjectSizeStruct = _propTypes.default.shape({
  xs: sizesStruct,
  md: sizesStruct,
  lg: sizesStruct,
  xl: sizesStruct
});
var propTypes = {
  /** Callback fired when the value is changed. */
  onChange: _propTypes.default.func,
  /** The value of the input element, required for a controlled component. */
  value: _propTypes.default.shape({
    firstName: _propTypes.default.string,
    lastName: _propTypes.default.string,
    id: _propTypes.default.string,
    birthDate: _propTypes.default.string
  }),
  /** The grid items sizing */
  grid: _propTypes.default.shape({
    bank: singleObjectSizeStruct,
    branch: singleObjectSizeStruct,
    accountNumber: singleObjectSizeStruct
  }),
  /** The grid container spacing */
  spacing: _propTypes.default.number,
  /** Name attribute of the input element. */
  name: _propTypes.default.string,
  /**    The helper text content. */
  helperText: _propTypes.default.shape({
    bank: _propTypes.default.string,
    branch: _propTypes.default.string,
    accountNumber: _propTypes.default.string
  }),
  /** If true, the label is displayed as required and the input element` will be required. */
  required: _propTypes.default.bool,
  /** If true, the label will be displayed in an error state. */
  error: _propTypes.default.shape({
    bank: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string]),
    branch: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string]),
    accountNumber: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string])
  }),
  /** DEV ONLY FLAG!! Pass proxy server url to allow on local */
  proxy: _propTypes.default.string
};
var defaultProps = {
  onChange: function onChange(e) {},
  spacing: 2,
  value: {
    firstName: '',
    lastName: '',
    id: '',
    birthDate: ''
  },
  grid: {
    firstName: {
      xs: 6
    },
    lastName: {
      xs: 6
    },
    id: {
      xs: 6
    },
    birthDate: {
      xs: 6
    }
  },
  name: 'Identification'
  // helperText: {},
  // error: {}
};

IdentificationInner.propTypes = _objectSpread(_objectSpread({}, propTypes), _useReduxField.propTypes);
IdentificationInner.defaultProps = _objectSpread({}, defaultProps);
var Identification = function Identification(props) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, (0, _useReduxField.default)(IdentificationInner)(props));
};
Identification.propTypes = _objectSpread(_objectSpread({}, propTypes), _useReduxField.propTypes);
Identification.defaultProps = _objectSpread({}, defaultProps);
Identification.displayName = 'Identification';
var _default = Identification;
exports.default = _default;
