// common-ui/lib/components/CreditCardField/CreditCardField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _TextField = _interopRequireDefault(require("../TextField"));
var _core = require("@material-ui/core");
var _CreditCardField = _interopRequireDefault(require("./CreditCardField.images"));
var _CreditCardField2 = require("./CreditCardField.formatter");
var _CreditCardField3 = _interopRequireDefault(require("./CreditCardField.styles"));
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _excluded = ["value", "onChange", "className", "hasSpaces"];
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var CreditCardField = function CreditCardField(props) {
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var value = props.value,
    onChange = props.onChange,
    _props$className = props.className,
    className = _props$className === void 0 ? null : _props$className,
    _props$hasSpaces = props.hasSpaces,
    hasSpaces = _props$hasSpaces === void 0 ? false : _props$hasSpaces,
    rest = (0, _objectWithoutProperties2.default)(props, _excluded);
  var delimiter = hasSpaces ? ' ' : '';
  var _useState = (0, _react.useState)(_CreditCardField.default.placeholder),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    ccIssuerImage = _useState2[0],
    setCCIssuerImage = _useState2[1];
  var _useState3 = (0, _react.useState)(''),
    _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
    altDescription = _useState4[0],
    setAltDescription = _useState4[1];
  var valueProps = (0, _CreditCardField2.formatCardNumber)(value, delimiter) || '';
  var _useState5 = (0, _react.useState)(valueProps),
    _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
    ccNumber = _useState6[0],
    setCCNumber = _useState6[1];
  var classes = (0, _CreditCardField3.default)();
  (0, _react.useEffect)(function () {
    setCCNumber(valueProps);
  }, [valueProps]);
  (0, _react.useEffect)(function () {
    var ccType = (0, _CreditCardField2.getCardTypeByValue)(ccNumber);
    var ccImage = ccType ? ccType.image : _CreditCardField.default.placeholder;
    var altDescription = ccType ? ccType === null || ccType === void 0 ? void 0 : ccType.type : '';
    setCCIssuerImage(ccImage);
    setAltDescription(altDescription);
  }, [ccNumber]);
  var fieldOnChange = function fieldOnChange(e) {
    var formatterValue = (0, _CreditCardField2.formatCardNumber)(e.target.value, delimiter);
    e.target.value = formatterValue;
    setCCNumber(formatterValue);
    onChange && onChange(e);
  };
  return /*#__PURE__*/_react.default.createElement(_TextField.default, (0, _extends2.default)({
    mask: hasSpaces ? '9999 9999 9999 9999' : '9999999999999999',
    value: ccNumber,
    onChange: fieldOnChange,
    invisibleRequired: true,
    type: "tel",
    directionLtr: true
  }, rest, {
    inputProps: _objectSpread(_objectSpread({}, props === null || props === void 0 ? void 0 : props.inputProps), {}, {
      className: className
    }),
    InputProps: {
      startAdornment: ccNumber && /*#__PURE__*/_react.default.createElement(_core.InputAdornment, {
        position: "start"
      }, /*#__PURE__*/_react.default.createElement("img", {
        style: {
          width: 30,
          height: 20
        },
        src: ccIssuerImage,
        draggable: "false",
        className: classes.iconCard,
        alt: altDescription
      }))
    }
  }));
};
CreditCardField.propTypes = {
  value: _propTypes.default.any,
  onChange: _propTypes.default.func,
  hasSpaces: _propTypes.default.bool
};
CreditCardField.defaultProps = {
  'data-hrl-bo': 'CreditCardField'
};
var _default = CreditCardField;
exports.default = _default;
