// common-ui/lib/components/BaseLayoutProvider/Themes/side/side.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var _base = _interopRequireDefault(require("../base"));
var _utils = require("../../../../utils");
require("../base/styles/font.css");
require("../base/styles/base.css");
/**
 * Side theme (switch look).
 * Put form on right side of screen (1/3)
 * and visual element on left side of screen (2/3)
 */

var theme = {
  palette: {
    primary: {
      main: '#2d76cb'
    },
    secondary: {
      main: '#f7b207'
    },
    error: {
      main: '#db2828'
    },
    background: {
      default: '#fff'
    },
    text: {
      primary: 'rgba(0, 63, 128, 1)',
      // 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 63, 128, 0.54)',
      disabled: 'rgba(0, 63, 128, 0.38)',
      hint: 'rgba(0, 63, 128, 0.38)'
    },
    divider: 'rgba(0, 0, 0, 0.12)'
  },
  overrides: {
    MuiButton: {
      label: {
        lineHeight: 1
      },
      root: {
        '&:not(:last-child)': {
          marginLeft: '16px'
        },
        fontSize: '1rem',
        borderRadius: '5rem',
        padding: '16px 32px',
        minWidth: '120px',
        '&:focus': {
          boxShadow: '0 0 10px #000 !important',
          border: '1px solid rgba(255,255,255,0.5)'
        }
      },
      contained: {
        boxShadow: 'none',
        '&:active': {
          boxShadow: 'none'
        },
        '&$focusVisible': {
          boxShadow: 'none'
        }
      },
      outlined: {
        padding: '16px 32px',
        background: '#fff',
        color: '#2d76cb',
        '& span': {
          color: '#2d76cb'
        }
      },
      text: {
        padding: '16px 32px'
      }
    }
  }
};
var sideTheme = (0, _utils.deepmerge)(_base.default, theme);
var _default = (0, _styles.createMuiTheme)(sideTheme);
exports.default = _default;
