// common-ui/lib/components/Strip/Strip.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    strip: {
      '& + &': {
        marginTop: theme.spacing(1)
      }
    },
    header: function header(props) {
      return {
        cursor: !props.isOnlyButtonOpen ? 'pointer' : 'auto',
        background: '#E3EBF6',
        padding: '12px 16px',
        position: 'relative',
        borderTopRightRadius: '4px',
        borderTopLeftRadius: '4px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between'
        // '& > span': {
        //     flex: 1,
        // },
      };
    },

    withCollapse: function withCollapse(props) {
      return {
        paddingLeft: props.arrowPosition === 'left' ? '46px' : '16px',
        paddingRight: props.arrowPosition === 'right' ? '46px' : '16px'
      };
    },
    iconCollapseOpen: {
      transform: 'translateY(-50%) rotate(0deg) !important'
    },
    iconCollapse: function iconCollapse(props) {
      return {
        position: 'absolute',
        padding: '0',
        left: props.arrowPosition === 'left' ? '8px' : 'auto',
        right: props.arrowPosition === 'right' ? '8px' : 'auto',
        top: '50%',
        transform: 'translateY(-50%) rotate(180deg)',
        marginLeft: 'auto',
        transition: theme.transitions.create('transform', {
          duration: theme.transitions.duration.shortest
        }),
        zIndex: 10
      };
    },
    actions: {
      display: 'flex',
      paddingRight: theme.spacing(1),
      '& > .MuiIconButton-root': {
        padding: 0
      },
      zIndex: 10,
      position: 'relative'
    },
    content: {
      background: '#fff',
      padding: '16px',
      borderBottomRightRadius: '4px',
      borderBottomLeftRadius: '4px'
    },
    clickableHeader: {
      position: 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      zIndex: 0
    }
  };
});
var _default = useStyles;
exports.default = _default;
