// common-ui/lib/components/ActionsBar/ActionsBar.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      direction: 'ltr',
      display: 'flex',
      '& svg': {
        fontSize: '24px'
      },
      '& > button': {
        backgroundColor: '#fff',
        width: '46px',
        height: '36px',
        cursor: 'pointer',
        border: 'none',
        display: 'inline-flex',
        alignSelf: 'center',
        justifyContent: 'center',
        color: theme.palette.secondary.main,
        overflow: 'visible',
        '&:hover, &:focus, &.active': {
          color: theme.palette.primary.main
        },
        '&:first-child': {
          borderBottomLeftRadius: '16px',
          borderTopLeftRadius: '16px'
        },
        '&:last-child': {
          borderBottomRightRadius: '16px',
          borderTopRightRadius: '16px'
        },
        '&:disabled': {
          color: '#ccc',
          cursor: 'not-allowed'
        }
      },
      '& > button + button': {
        borderLeft: '1px solid #e3ebf6'
      }
    }
  };
});
var _default = useStyles;
exports.default = _default;
