// common-ui/lib/components/ActionsBar/Action.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Tooltip = _interopRequireDefault(require("../Tooltip"));

var _ = require("../..");

var noop = function noop() {};

var Action = function Action(props) {
  var alt = props.alt,
      icon = props.icon,
      disabled = props.disabled,
      onClick = props.onClick,
      name = props.name,
      hrlbo = props.hrlbo,
      badgeProps = props.badgeProps;

  var handleClick = function handleClick(e) {
    onClick(e, name);
  };

  var Icon = /*#__PURE__*/_react.default.createElement(_.Badge, badgeProps, icon);

  var action = /*#__PURE__*/_react.default.createElement("button", {
    "aria-label": alt,
    name: name,
    "data-hrl-bo": hrlbo,
    disabled: disabled,
    onClick: handleClick
  }, Icon);

  return alt ? /*#__PURE__*/_react.default.createElement(_Tooltip.default, {
    title: alt
  }, action) : action;
};

Action.propTypes = {
  /**
   * Alt text to be visible with tooltip
   */
  alt: _propTypes.default.string,

  /**
   * Icon element to be visible
   */
  icon: _propTypes.default.node.isRequired,

  /**
   * If true, action is disabled
   */
  disabled: _propTypes.default.bool,

  /**
   * OnClick handler function
   */
  onClick: _propTypes.default.func,

  /**
   * Action name
   */
  name: _propTypes.default.string,

  /**
   * data-hrl-bo attribute value
   */
  hrlbo: _propTypes.default.string,

  /**
   * Props for badge component. See props table in badge component
   */
  badgeProps: _propTypes.default.shape({
    classes: _propTypes.default.object,
    position: _propTypes.default.object,
    badgeContent: _propTypes.default.node,
    max: _propTypes.default.number,
    variant: _propTypes.default.oneOf(['standard', 'dot']),
    color: _propTypes.default.oneOf(['default', 'error', 'primary', 'secondary'])
  })
};
Action.defaultProps = {
  onClick: noop,
  badgeProps: {
    badgeContent: 0
  }
};
var _default = Action;
exports.default = _default;
