// harelkit/lib/components/LayoutProvider/Layouts/default/default.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _cloud = _interopRequireDefault(require("./assets/images/cloud.svg"));

var _skyline = _interopRequireDefault(require("./assets/images/skyline.svg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  position: absolute;\n  background-repeat: repeat-x;\n  background-size: cover;\n  z-index: -1;\n  background-image: url(", ");\n  bottom: 0;\n  height: 129px;\n  left: 0;\n  right: 0;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  top: 400px;\n  height: 20px;\n  width: 46px;\n  right: 180px;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  top: 260px;\n  height: 26px;\n  width: 57px;\n  left: 180px;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  top: 120px;\n  height: 45px;\n  width: 100px;\n  left: 220px;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  position: absolute;\n  background-repeat: no-repeat;\n  background-size: cover;\n  opacity: 0.9;\n  background-image: url(", ");\n  z-index: -2;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  width: 100%;\n  overflow: hidden;\n  position: absolute;\n  top: 0;\n  right: 0;\n  bottom: 0;\n  left: 0;\n  z-index: -1;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  min-height: 100vh;\n  background-image: linear-gradient(to bottom, #c9e2ff 19%, #f3f3f3 72%);\n  z-index: 1;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Layout = _styledComponents.default.div(_templateObject());

var Images = _styledComponents.default.div(_templateObject2());

var Cloud = _styledComponents.default.div(_templateObject3(), _cloud.default);

var Cloud1 = (0, _styledComponents.default)(Cloud)(_templateObject4());
var Cloud2 = (0, _styledComponents.default)(Cloud)(_templateObject5());
var Cloud3 = (0, _styledComponents.default)(Cloud)(_templateObject6());

var Foot = _styledComponents.default.div(_templateObject7(), _skyline.default);

var _default = {
  Layout: Layout,
  Images: Images,
  Cloud: Cloud,
  Cloud1: Cloud1,
  Cloud2: Cloud2,
  Cloud3: Cloud3,
  Foot: Foot
};
exports.default = _default;
