// harelkit/lib/components/DatePicker/DatePicker.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _DatePicker = _interopRequireDefault(require("./DatePicker.style"));

var _Input = _interopRequireDefault(require("./Input"));

var _Header = _interopRequireDefault(require("./Header"));

var _reactDatepicker = _interopRequireWildcard(require("react-datepicker"));

require("react-datepicker/dist/react-datepicker.css");

var _he = _interopRequireDefault(require("date-fns/locale/he"));

var _format = _interopRequireDefault(require("date-fns/format"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

(0, _reactDatepicker.registerLocale)('he', _he.default);
var dateFormat = "dd/MM/yyyy";

var noop = function noop() {};

var DatePicker = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var id = props.id,
      name = props.name,
      onOpen = props.onOpen,
      onChange = props.onChange,
      disabled = props.disabled,
      excludeDates = props.excludeDates,
      inline = props.inline,
      label = props.label,
      placeholder = props.placeholder,
      minDate = props.minDate,
      maxDate = props.maxDate,
      date = props.date,
      openToDate = props.openToDate,
      bo = props.bo,
      helperText = props.helperText,
      error = props.error,
      startYear = props.startYear,
      endYear = props.endYear,
      closeOnScroll = props.closeOnScroll,
      required = props.required,
      rest = _objectWithoutProperties(props, ["id", "name", "onOpen", "onChange", "disabled", "excludeDates", "inline", "label", "placeholder", "minDate", "maxDate", "date", "openToDate", "bo", "helperText", "error", "startYear", "endYear", "closeOnScroll", "required"]);

  var datepickerRef = /*#__PURE__*/(0, _react.createRef)();
  var selectedDate = date instanceof Date ? date : null;

  var _useState = (0, _react.useState)(selectedDate),
      _useState2 = _slicedToArray(_useState, 2),
      startDate = _useState2[0],
      setStartDate = _useState2[1];

  var inputProps = {
    bo: bo,
    id: id,
    label: label,
    required: required,
    onOpen: onOpen,
    placeholder: placeholder,
    disabled: disabled,
    helperText: helperText,
    error: error,
    name: name
  };
  var headerProps = {
    startYear: startYear,
    endYear: endYear
  };

  var handleDateChange = function handleDateChange(date) {
    // onChange(date);
    setStartDate(date);
  };

  var handleInputChange = function handleInputChange(e) {
    onChange(startDate);
  };

  var CustomInput = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
    var value = _ref.value,
        onClick = _ref.onClick;

    var handleButtonOnClick = function handleButtonOnClick() {
      if (onOpen) onOpen();
      if (onClick) onClick();
    };

    return /*#__PURE__*/_react.default.createElement("button", {
      onClick: handleButtonOnClick,
      "aria-label": "\u05E4\u05EA\u05D9\u05D7\u05EA \u05EA\u05D0\u05E8\u05D9\u05DB\u05D5\u05DF"
    }, /*#__PURE__*/_react.default.createElement("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 16 15"
    }, /*#__PURE__*/_react.default.createElement("g", {
      fill: "currentColor",
      fillRule: "evenodd"
    }, /*#__PURE__*/_react.default.createElement("path", {
      stroke: "currentColor",
      strokeWidth: ".2",
      d: "M.74 13.765c0 .342.3.63.655.63h12.922c.355 0 .654-.288.654-.63V5.891H.741v7.874zm.655 1.15c-.66 0-1.195-.516-1.195-1.15V2.63c0-.633.536-1.149 1.195-1.149h2.582l.004.52H1.395c-.355 0-.655.288-.655.63v2.74h14.231V2.63c0-.342-.3-.63-.654-.63H3.981v-.52h10.336c.659 0 1.195.516 1.195 1.15v11.134c0 .634-.536 1.15-1.195 1.15H1.395z"
    }), /*#__PURE__*/_react.default.createElement("path", {
      d: "M11.08 3.132V.237c0-.131.135-.237.302-.237.166 0 .302.106.302.236v2.896c0 .13-.136.236-.302.236-.167 0-.303-.106-.303-.236zm-3.83 0V.237c0-.131.135-.237.302-.237.166 0 .302.106.302.236v1.11l.002.473-.002.199v1.114c0 .13-.136.236-.302.236-.167 0-.303-.106-.303-.236zm-3.75 0V.237C3.5.105 3.636 0 3.802 0c.167 0 .302.106.302.236v1.11l.003.473-.003.199v1.114c0 .13-.135.236-.302.236-.166 0-.302-.106-.302-.236zM.2 6h15v8a1 1 0 01-1 1h-13a1 1 0 01-1-1V6z"
    }))));
  });
  return /*#__PURE__*/_react.default.createElement(_DatePicker.default.Datepicker, null, !inline && /*#__PURE__*/_react.default.createElement(_Input.default, _extends({}, rest, inputProps, {
    value: startDate && (0, _format.default)(startDate, dateFormat) || "",
    placeholder: dateFormat,
    onChange: handleInputChange,
    error: error,
    helperText: helperText,
    ref: ref
  })), /*#__PURE__*/_react.default.createElement(_reactDatepicker.default, {
    selected: startDate,
    onChange: handleDateChange,
    ref: datepickerRef,
    customInput: /*#__PURE__*/_react.default.createElement(CustomInput, null),
    showPopperArrow: false,
    popperPlacement: "bottom-start",
    locale: "he",
    dateFormat: dateFormat,
    disabled: disabled,
    excludeDates: excludeDates,
    inline: inline,
    minDate: minDate,
    maxDate: maxDate,
    openToDate: openToDate,
    closeOnScroll: closeOnScroll,
    renderCustomHeader: function renderCustomHeader(props) {
      return /*#__PURE__*/_react.default.createElement(_Header.default, _extends({}, headerProps, props));
    }
  }));
});
DatePicker.propTypes = {
  /**
   * Datepicker Text field id
   */
  id: _propTypes.default.string,

  /**
   * Daetpicker Text field name
   */
  name: _propTypes.default.string,

  /**
   * Callback fired when the datepicker open.
   * function() => void
   */
  onOpen: _propTypes.default.func,

  /**
   * Callback fired when the datepicker changes date.
   * function(date) => void
   */
  onChange: _propTypes.default.func,

  /**
   * If true, the input is disabled
   */
  disabled: _propTypes.default.bool,
  // excludeDates,

  /**
   * If true, datepicker is inline on screen
   */
  inline: _propTypes.default.bool,

  /**
   * Datepicker Textfield label
   */
  label: _propTypes.default.string,

  /**
   * Datepicker Textfield placeholder
   */
  placeholder: _propTypes.default.string,

  /**
   * Datepicker Textfield helper text
   */
  helperText: _propTypes.default.string,

  /**
   * If true, Datepicker Textfield change the colors to error
   */
  error: _propTypes.default.bool,

  /**
   * Minimum date 
   */
  minDate: _propTypes.default.instanceOf(Date),

  /**
   * Maximum date 
   */
  maxDate: _propTypes.default.instanceOf(Date),

  /**
   * Datepicker start date
   */
  date: _propTypes.default.instanceOf(Date),

  /**
   * Year to start from in years dropdown
   */
  startYear: _propTypes.default.number,

  /**
   * Year to end in years dropdown
   */
  endYear: _propTypes.default.number,

  /**
   * If true, close datepicker on screen scroll
   */
  closeOnScroll: _propTypes.default.bool,

  /**
   * If set, asterix sign will be added.
   * this does not set validation rule
   */
  required: _propTypes.default.bool,

  /**
   * Will add data-hrl-bo attribute to element.
   * 
   * bo attribute:
   * {bo}-datepicker-textfield
   * {bo}-datepicker-textfield-label
   * {bo}-datepicker-textfield-help
   */
  bo: _propTypes.default.string
};
DatePicker.defaultProps = {
  onOpen: noop,
  onChange: noop,
  disabled: false,
  error: false,
  inline: false,
  startYear: 2000,
  closeOnScroll: true,
  required: false
};
var _default = DatePicker;
exports.default = _default;
