// harelkit/lib/components/DatePicker/Header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _dateFns = require("date-fns");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var range = function range(start, end) {
  return Array(end - start + 1).fill().map(function (_, idx) {
    return start + idx;
  });
};

var months = ['ינואר', 'פברואר', 'מרץ', 'אפריל', 'מאי', 'יוני', 'יולי', 'אוגוסט', 'ספטמבר', 'אוקטובר', 'נובמבר', 'דצמבר'];

var Header = function Header(props) {
  var startYear = props.startYear,
      endYear = props.endYear,
      date = props.date,
      changeYear = props.changeYear,
      changeMonth = props.changeMonth,
      decreaseMonth = props.decreaseMonth,
      increaseMonth = props.increaseMonth,
      prevMonthButtonDisabled = props.prevMonthButtonDisabled,
      nextMonthButtonDisabled = props.nextMonthButtonDisabled;
  var years = range(startYear, endYear || new Date().getFullYear() + 1);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("button", {
    onClick: decreaseMonth,
    disabled: prevMonthButtonDisabled
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 16 16"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M5 .929l6.238 7.07L5 15.072",
    stroke: "currentColor",
    strokeWidth: "2",
    fill: "none",
    fillRule: "evenodd",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }))), /*#__PURE__*/_react.default.createElement("select", {
    value: months[(0, _dateFns.getMonth)(date)],
    onChange: function onChange(_ref) {
      var value = _ref.target.value;
      return changeMonth(months.indexOf(value));
    }
  }, months.map(function (option) {
    return /*#__PURE__*/_react.default.createElement("option", {
      key: option,
      value: option
    }, option);
  })), /*#__PURE__*/_react.default.createElement("select", {
    value: (0, _dateFns.getYear)(date),
    onChange: function onChange(_ref2) {
      var value = _ref2.target.value;
      return changeYear(value);
    }
  }, years.map(function (option) {
    return /*#__PURE__*/_react.default.createElement("option", {
      key: option,
      value: option
    }, option);
  })), /*#__PURE__*/_react.default.createElement("button", {
    onClick: increaseMonth,
    disabled: nextMonthButtonDisabled
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 16 16"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M11 .929L4.762 8 11 15.071",
    stroke: "currentColor",
    strokeWidth: "2",
    fill: "none",
    fillRule: "evenodd",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }))));
};

var _default = Header;
exports.default = _default;
