// harelkit/lib/components/ThemeProvider/Themes/base/overrides/MuiButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _palette = _interopRequireDefault(require("../palette"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  root: {
    textTransform: null,
    borderRadius: '24px',
    lineHeight: '1.1',
    padding: '9px 20px',
    transition: 'transform 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    '&:active': {
      transform: 'scale(0.95)'
    },
    '&.Mui-focusVisible:before': {
      content: '""',
      display: 'block',
      border: "1px solid ".concat(_palette.default.primary.main),
      position: 'absolute',
      top: '-4px',
      right: '-4px',
      left: '-4px',
      bottom: '-4px',
      borderRadius: '24px'
    }
  },
  containedSecondary: {
    '&.Mui-focusVisible:before': {
      borderColor: _palette.default.secondary.main
    }
  },
  startIcon: {
    marginLeft: '8px',
    marginRight: '-4px'
  },
  endIcon: {
    marginLeft: '-4px',
    marginRight: '8px'
  },
  disabled: {},
  text: {
    padding: null
  },
  outlined: {
    padding: null
  },
  contained: {
    boxShadow: null,
    '&:hover': {
      boxShadow: null
    }
  },
  containedSizeLarge: {
    padding: '15px 30px'
  },
  outlinedSizeLarge: {
    padding: '15px 30px'
  },
  textSizeLarge: {
    padding: '15px 30px'
  }
};
exports.default = _default;
