// harelkit/lib/components/ThemeProvider/Themes/base/overrides/MuiTypography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
//TODO: find a way to use breakpoints value instead of hardcore value
var MuiTypography = {
  root: {
    margin: '0'
  },
  gutterBottom: {
    marginBottom: '0.5em'
  },
  h1: {
    fontSize: '3.125rem',
    lineHeight: '1.2'
  },
  h2: {
    fontSize: '2.25rem',
    lineHeight: '1.16666666667',
    '@media (max-width:518px)': {
      fontSize: '2rem'
    }
  },
  h3: {
    fontSize: '1.5rem',
    lineHeight: '1.125'
  },
  h4: {
    fontSize: '1.25rem',
    fontWeight: 'bold',
    lineHeight: '1.25'
  },
  h5: {
    fontSize: '1.25rem',
    lineHeight: '1.25'
  },
  h6: {
    fontSize: '1.25rem',
    fontWeight: '300',
    lineHeight: '1.25'
  },
  subtitle1: {
    fontSize: '1.5rem',
    lineHeight: '1.25',
    '@media (max-width:518px)': {
      fontSize: '1.25rem'
    }
  },
  subtitle2: {
    fontSize: '1.25rem',
    lineHeight: '1.25',
    '@media (max-width:518px)': {
      fontSize: '1rem'
    }
  },
  body1: {
    fontSize: '1rem',
    lineHeight: '1.25',
    color: '#003c7f'
  },
  body2: {
    fontSize: '0.875rem',
    lineHeight: '1.14285714286',
    color: '#003c7f'
  },
  caption: {
    fontSize: '1rem',
    lineHeight: '1.08333333333',
    '@media (max-width:518px)': {
      fontSize: '0.75rem'
    }
  },
  button: {
    fontSize: '1rem',
    fontWeight: 'bold'
  }
};
var _default = MuiTypography;
exports.default = _default;
