// common-ui/lib/components/InfoBlock/InfoBlock.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: function root(props) {
      return {
        position: 'relative',
        padding: theme.spacing(1),
        borderRadius: theme.spacing(0.5),
        color: '#444',
        // TODO: make this color part of theme pallete
        backgroundColor: '#eee',
        // TODO: make this color part of theme pallete
        '&:not(:last-child)': {
          marginBottom: theme.spacing(1)
        },
        paddingLeft: props.showClose ? theme.spacing(6) : null
      };
    },
    close: {
      position: 'absolute',
      top: theme.spacing(1),
      left: theme.spacing(1),
      padding: '4px',
      fontSize: '1rem'
    },
    mainText: {
      margin: '0',
      marginTop: theme.spacing(0.5)
    },
    error: {
      background: "".concat(theme.palette.error.light, " !important"),
      color: "".concat(theme.palette.error.lightText, " !important")
    },
    info: {
      background: "".concat(theme.palette.info.light, " !important"),
      color: "".concat(theme.palette.info.lightText, " !important")
    },
    success: {
      background: "".concat(theme.palette.success.light, " !important"),
      color: "".concat(theme.palette.success.lightText, " !important")
    },
    warning: {
      background: "".concat(theme.palette.warn.light, " !important"),
      color: "".concat(theme.palette.warn.contrastText, " !important")
    }
  };
});
var _default = useStyles;
exports.default = _default;
