// common-ui/lib/components/Select/GroupLine.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var useStyles = (0, _core.makeStyles)(function (theme) {
  return {
    root: {
      // margin: '2px 16px',
      //borderBottom: `1px solid ${theme.palette.primary.main}`,
      margin: '2px auto',
      maxWidth: 'calc(100% - 15px)',
      '& hr': {
        width: '100%',
        height: '1px',
        border: 'none',
        borderBottom: "1px solid ".concat(theme.palette.primary.main)
      }
    }
  };
});
var GroupLine = function GroupLine() {
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("li", {
    className: classes.root,
    role: "separator"
  }, /*#__PURE__*/_react.default.createElement("hr", {
    className: classes.root
  }));
};
GroupLine.displayName = 'GroupLine';
var _default = GroupLine;
exports.default = _default;
