// harelkit/lib/components/Header/Header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Header = _interopRequireDefault(require("./Header.style"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var noop = function noop() {};

var Header = function Header(props) {
  var showLogo = props.showLogo,
      onClose = props.onClose,
      onLogoClick = props.onLogoClick,
      title = props.title,
      showClose = props.showClose;

  var handleLogoClick = function handleLogoClick(e) {
    onLogoClick();
  };

  var handleCloseClick = function handleCloseClick(e) {
    onClose();
  };

  return /*#__PURE__*/_react.default.createElement(_Header.default.Header, null, showLogo && /*#__PURE__*/_react.default.createElement(_Header.default.Logo, {
    onClick: handleLogoClick
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 61 22"
  }, /*#__PURE__*/_react.default.createElement("defs", null, /*#__PURE__*/_react.default.createElement("path", {
    id: "a",
    d: "M0 .073h8.325v14.063H0z"
  }), /*#__PURE__*/_react.default.createElement("path", {
    id: "c",
    d: "M.023.07h11.062v10.795H.023z"
  })), /*#__PURE__*/_react.default.createElement("g", {
    fill: "none",
    fillRule: "evenodd"
  }, /*#__PURE__*/_react.default.createElement("path", {
    fill: "#0088C9",
    d: "M53.166 2.879c2.562 1.846 2.01 4.99-.006 6.908-.171-1.342-.938-2.182-1.836-2.703-1.816-1.054-5.556-.936-8.338 1.946-.652.675-1.186 1.742-1.589 2.487-1.095-2.427-.297-5.084 1.991-7.453 2.93-3.035 7.096-3.119 9.778-1.185"
  }), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#F7B207",
    d: "M43.719 10.417c.375-3.065 3.47-4.176 6.199-3.45-1.118.816-1.488 1.877-1.506 2.889-.037 2.045 1.938 5.112 5.914 5.998.932.209 2.156.121 3.026.085-1.63 2.129-4.412 2.791-7.681 2.062-4.187-.933-6.345-4.375-5.952-7.584"
  }), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#2D9D3A",
    d: "M55.315 14.277c-2.936 1.22-6.888-1.366-6.594-4.456.845 1.06 2.488 1.505 3.744.813 1.84-1.013 3.616-4.178 2.424-7.945-.28-.883-.97-1.863-1.437-2.572 2.724.298 4.709 2.293 5.69 5.391 1.256 3.967-.751 7.493-3.827 8.77"
  }), /*#__PURE__*/_react.default.createElement("g", {
    transform: "translate(0 6.997)"
  }, /*#__PURE__*/_react.default.createElement("mask", {
    id: "b",
    fill: "#fff"
  }, /*#__PURE__*/_react.default.createElement("use", {
    xlinkHref: "#a"
  })), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#4F5254",
    d: "M0 .073v5.448h4.875L1.84 14.136h3.381L8.325 5.48V3.342H3.358V.073z",
    mask: "url(#b)"
  })), /*#__PURE__*/_react.default.createElement("g", {
    transform: "translate(9.196 10.268)"
  }, /*#__PURE__*/_react.default.createElement("mask", {
    id: "d",
    fill: "#fff"
  }, /*#__PURE__*/_react.default.createElement("use", {
    xlinkHref: "#c"
  })), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#4F5254",
    d: "M11.085 10.864S9.131 7.76 8.05 6.301c1.264-.391 2.645-1.152 2.668-3.229V.07H7.29v4.174c0 .596-.252.515-.252.515L4.14.07H.023l3.105 3.885C1.818 4.573.507 5.087.507 7.946v2.919h3.426V6.012c0-.556.368-.514.368-.514l3.013 5.365h3.771z",
    mask: "url(#d)"
  })), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#4F5254",
    d: "M28.307 21.133V13.73c0-2.652-1.08-3.392-3.633-3.392h-3.565v2.18h2.323c1.196-.021 1.426.349 1.426 1.397v7.217h3.45zm11.404 0V13.73c0-2.652-1.104-3.392-3.656-3.392h-5.566v2.18h4.347c1.173 0 1.426.37 1.426 1.418v7.196h3.45zm-5.841-6.476l-3.38.698v5.778h3.38v-6.476z"
  })))), title && /*#__PURE__*/_react.default.createElement(_Header.default.Title, null, title), /*#__PURE__*/_react.default.createElement(_Header.default.Actions, null, showClose && /*#__PURE__*/_react.default.createElement("button", {
    onClick: handleCloseClick
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 31 31"
  }, /*#__PURE__*/_react.default.createElement("path", {
    fill: "none",
    stroke: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    d: "M9.23 9.23l12.54 12.54m0-12.54L9.23 21.77"
  })))), /*#__PURE__*/_react.default.createElement(_Header.default.Bg, null, /*#__PURE__*/_react.default.createElement("svg", {
    focusable: "false",
    viewBox: "0 0 1920 70",
    "aria-hidden": "true",
    role: "presentation",
    width: "1920",
    height: "70"
  }, /*#__PURE__*/_react.default.createElement("defs", null, /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: "b",
    x1: "50%",
    x2: "50%",
    y1: "106.46%",
    y2: "0%"
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: "0%",
    stopColor: "#FCFAFA"
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: "53.414%",
    stopColor: "#FCFAFA"
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: "100%",
    stopColor: "#F3F5F5"
  })), /*#__PURE__*/_react.default.createElement("path", {
    id: "a",
    d: "M0 0h1920v70H0z"
  })), /*#__PURE__*/_react.default.createElement("g", {
    fill: "none",
    fillRule: "evenodd"
  }, /*#__PURE__*/_react.default.createElement("mask", {
    id: "c",
    fill: "#fff"
  }, /*#__PURE__*/_react.default.createElement("use", {
    xlinkHref: "#a"
  })), /*#__PURE__*/_react.default.createElement("use", {
    fill: "url(#b)",
    stroke: "#979797",
    strokeOpacity: ".196",
    xlinkHref: "#a"
  }), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#83878B",
    d: "M714.41 72.934C618.721-43.149 682.79-186.29 799.514-255c-8.54 65.639 16.583 114.015 51.699 148.724 70.974 70.176 245.224 106.067 409.196.135 38.446-24.815 76.214-69.566 103.968-100.48 20.76 127.4-48.7 244.72-183.543 331.823C1008.158 236.746 814.61 194.49 714.41 72.934",
    mask: "url(#c)",
    opacity: ".045"
  }), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#83878B",
    d: "M221.41 163.934C125.721 47.851 189.79-95.29 306.514-164c-8.54 65.639 16.583 114.015 51.699 148.724C429.188 54.9 603.438 90.791 767.41-15.14c38.446-24.815 76.214-69.566 103.968-100.48 20.76 127.4-48.7 244.72-183.543 331.823C515.158 327.746 321.61 285.49 221.41 163.934",
    mask: "url(#c)",
    opacity: ".045"
  }), /*#__PURE__*/_react.default.createElement("path", {
    fill: "#9A9EA2",
    d: "M1214.161 231.906c-134.098-70.127-131.716-227.122-51.366-335.781 18.107 63.92 60.398 98.858 106.487 117.307 93.16 37.309 267.904 2.956 377.053-158.376 25.602-37.806 42.688-93.76 56.025-133.056 69.544 109.601 51.976 244.84-37.788 377.502-114.948 169.886-309.99 205.837-450.41 132.404",
    mask: "url(#c)",
    opacity: ".045"
  })))));
};

Header.propTypes = {
  /**
   * If true, the logo is visible
   */
  showLogo: _propTypes.default.bool,

  /**
   * On closing from X click handler
   */
  onClose: _propTypes.default.func,

  /**
   * On logo click handler
   */
  onLogoClick: _propTypes.default.func,

  /**
   * Header title
   */
  title: _propTypes.default.string,

  /**
   * If true, the close button is visible
   */
  showClose: _propTypes.default.bool
};
Header.defaultProps = {
  showLogo: true,
  showClose: true,
  onClose: noop,
  onLogoClick: noop
};
var _default = Header;
exports.default = _default;
