// common-ui/lib/components/BaseLayoutProvider/Themes/next/next.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var _base = _interopRequireDefault(require("../base"));
var _utils = require("../../../../utils");
require("../base/styles/font.css");
require("../base/styles/base.css");
var theme = {
  direction: 'rtl',
  palette: {
    primary: {
      light: '#6099dc',
      main: '#2d76cb',
      dark: '#1e4e86',
      text: '#ffffff'
    },
    secondary: {
      light: '#f9c239',
      main: '#f7b207',
      dark: '#b68406',
      text: '#003e7a'
    },
    tertiary: {
      light: '#33b242',
      main: '#278732',
      dark: '#22772c',
      text: '#ffffff'
    }
  },
  overrides: {
    MuiAutocomplete: {
      input: {
        padding: '4px !important'
      }
    },
    MuiButton: {
      root: {
        padding: '15px 30px',
        fontSize: '16px',
        opacity: '100%',
        lineHeight: '18px',
        borderRadius: '24px',
        fontFamily: '"mfw_protocolharel", Arial',
        fontWeight: 'bold',
        position: 'relative',
        '&:focus': {
          borderColor: '#ffffff'
        },
        '&:focus::after': {
          content: '""',
          position: 'absolute',
          top: '-4px',
          left: '-4px',
          width: 'calc(100% + 8px)',
          height: 'calc(100% + 8px)',
          border: '1px solid #002667',
          borderRadius: '28px',
          display: 'block'
        },
        '&:hover': {
          boxShadow: 'none'
        },
        '&:visited': {}
      },
      text: {
        backgroundColor: 'transparent',
        padding: '15px 30px'
      },
      contained: {
        '&.Mui-disabled': {
          boxShadow: '0px 2px 4px -1px rgba(0,0,0,0.2)'
        },
        '&:hover': {
          boxShadow: 'none'
        }
      },
      outlined: {
        padding: '15px 30px'
      }
    },
    MuiListItem: {
      root: {
        '&.Mui-selected': {
          backgroundColor: 'transparent',
          fontWeight: 'bold'
        }
      }
    },
    MuiMenuItem: {
      root: {
        fontSize: '16px',
        lineHeight: '16px',
        paddingTop: '9px',
        paddingBottom: '9px'
      }
    },
    MuiFormlabel: {
      root: {
        paddingRight: 0
      }
    },
    MuiFormControl: {
      root: {
        borderRadius: 5,
        '& label.MuiFormLabel-root': {
          fontSize: '1rem',
          fontWeight: 400
        },
        '&.MuiFormControl-Select': {
          '& label': {
            right: '4px',
            fontSize: '16px',
            lineHeight: '18px',
            fontWeight: '500'
          },
          '& label.MuiInputLabel-shrink': {
            transform: 'translate(0, 6px) scale(0.85)',
            width: '110%'
          },
          '& label + .MuiInput-formControl': {
            marginTop: 0
          }
        },
        '&.MuiRadioGroup-root label': {
          marginLeft: 'inherit',
          marginRight: 'inherit'
        }
      }
    },
    MuiSelect: {
      filled: {
        paddingRight: 8
      },
      select: {
        '&.MuiSelect-selectMenu': {
          paddingBottom: '8px',
          paddingTop: '22px',
          paddingLeft: '6px',
          paddingRight: 8,
          borderRadius: '6px 6px 0 0',
          height: '16px',
          lineHeight: '18px',
          fontSize: '16px',
          fontWeight: '500'
        }
      },
      icon: {
        top: 'calc(50% - 8px)',
        left: 7,
        color: '#003e7a',
        fontSize: 16,
        fontWeight: 100
      },
      iconFilled: {
        left: 7,
        right: 'auto'
      }
    },
    MuiPaper: {
      rounded: {
        borderRadius: '0 0 4px 4px'
      },
      elevation8: {
        boxShadow: '0 1px 6px rgba(0,0,0,0.2)'
      }
    },
    MuiRadio: {
      root: {
        marginLeft: '8px',
        padding: '6px',
        border: 0,
        '& svg': {
          width: '22px',
          height: '22px'
        }
      }
    },
    MuiCheckbox: {
      root: {
        marginLeft: '8px',
        padding: '6px',
        border: 0,
        '& svg': {
          width: '22px',
          height: '22px'
        }
      }
    },
    MuiFilledInput: {
      root: {
        backgroundColor: '#e2ecf9',
        '&:hover': {
          backgroundColor: '#f2f7fc'
        },
        '&.Mui-focused': {
          backgroundColor: '#e2ecf9'
        }
      },
      input: {
        backgroundColor: 'transparent',
        padding: '23px 8px 4px'
      },
      underline: {
        '&.Mui-disabled:before': {
          borderBottom: '0px'
        }
      }
    },
    MuiInputLabel: {
      formControl: {
        right: 0
      },
      asterisk: {
        paddingLeft: '3px',
        color: _base.default.palette.error.main
      },
      filled: {
        transform: 'translate(-8px, 20px) scale(1)',
        '&.MuiInputLabel-shrink': {
          transform: 'translate(-8px, 10px) scale(0.75)',
          width: '125%'
        }
      }
    },
    MuiInputBase: {
      input: {
        fontSize: '16px',
        lineHeight: '18px'
      }
    },
    MuiFormHelperText: {
      contained: {
        margin: '4px 8px 0'
      }
    }
  }
};
var nextTheme = (0, _utils.deepmerge)(_base.default, theme);
var _default = (0, _styles.createMuiTheme)(nextTheme);
exports.default = _default;
