// common-ui/lib/components/Toggle/Option.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var mainBGColor = '#2d76cb';
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    input: {
      position: 'absolute',
      right: 0,
      opacity: 0,
      width: 'calc(100% + 2px)',
      height: 'calc(100% + 2px)',
      margin: 0,
      transition: 'all 0.2s',
      marginRight: '-1px',
      cursor: 'pointer',
      '&:focus -+ label': {
        boxShadow: '0px 0px 0px 2px rgba(0, 108, 178, 0.4)',
        zIndex: '1',
        borderColor: '#084C4',
        backgroundColor: '#C7E1FF'
      }
    },
    label: {
      userSelect: 'none',
      cursor: 'pointer',
      color: theme.palette.primary.secondary,
      display: 'flex',
      height: '100%',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: '#fff',
      border: "1px solid ".concat(mainBGColor),
      borderLeft: 'none',
      padding: theme.spacing(1.5),
      '&:hover': {
        backgroundColor: '#eaf1fa'
      },
      '&:active, &:focused': {
        backgroundColor: '#b2ceee'
      },
      '&[aria-checked="true"]': {
        backgroundColor: mainBGColor,
        color: '#fff'
      },
      '&:first-of-type': {
        borderTopRightRadius: theme.spacing(2),
        borderBottomRightRadius: theme.spacing(2),
        border: "1px solid ".concat(mainBGColor)
      },
      '&:last-of-type': {
        borderTopLeftRadius: theme.spacing(2),
        borderBottomLeftRadius: theme.spacing(2),
        border: "1px solid ".concat(mainBGColor)
      },
      '&:focus-within': {
        boxShadow: '0px 0px 0px 2px rgba(0, 108, 178, 0.4)',
        zIndex: '1'
      },
      '&[aria-disabled="true"]': {
        background: 'rgba(228, 228, 228, 0.5)',
        color: 'rgba(0, 0, 0, 0.3)',
        cursor: 'not-allowed'
      },
      '&[aria-checked="true"][aria-disabled="true"]': {
        background: 'rgba(228, 228, 228, 1)'
      },
      '[aria-invalid="true"] &': {
        borderColor: '#DB2A1D'
      }
    }
  };
});
var _default = useStyles;
exports.default = _default;
