// common-ui/lib/components/BaseLayoutProvider/BaseLayoutProvider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _styles = require("@material-ui/core/styles");
var _PickerWrappers = _interopRequireDefault(require("../PickerWrappers"));
var _default2 = _interopRequireDefault(require("./Themes/default"));
var _side = _interopRequireDefault(require("./Themes/side"));
var _next = _interopRequireDefault(require("./Themes/next"));
var _default3 = _interopRequireDefault(require("./Layouts/default"));
var _side2 = _interopRequireDefault(require("./Layouts/side"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var LayoutWrapper = function LayoutWrapper(_ref) {
  var Layout = _ref.Layout,
    layout = _ref.layout,
    children = _ref.children;
  // empty, null or undefined
  if (Layout == null) {
    return children;
  }

  // object
  if (typeof Layout !== 'string' && !(Layout instanceof String)) {
    return /*#__PURE__*/_react.default.createElement(Layout, null, children);
  }

  // string
  if (layout === 'side') {
    return /*#__PURE__*/_react.default.createElement(_side2.default, null, children);
  } else {
    return /*#__PURE__*/_react.default.createElement(_default3.default, null, children);
  }
};
LayoutWrapper.propTypes = {
  Layout: _propTypes.default.any,
  layout: _propTypes.default.any,
  children: _propTypes.default.node
};
var BaseLayoutProvider = function BaseLayoutProvider(props) {
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var Layout = props.Layout,
    Theme = props.Theme,
    theme = props.theme,
    layout = props.layout,
    children = props.children;

  // TODO: remove code when deprecating Layout & Theme
  if (theme) Theme = theme;
  if (layout) Layout = layout;
  var getTheme = function getTheme() {
    // empty, null or undefined
    if (Theme == null) {
      return _default2.default;
    }

    // object
    if (typeof Theme !== 'string' && !(Theme instanceof String)) {
      return Theme;
    }

    // string
    switch (Theme) {
      case 'side':
        return _side.default;
      case 'next':
        return _next.default;
      default:
        return _default2.default;
    }
  };
  var pageTheme = getTheme();
  return /*#__PURE__*/_react.default.createElement(_PickerWrappers.default, null, /*#__PURE__*/_react.default.createElement(_styles.ThemeProvider, {
    theme: pageTheme
  }, /*#__PURE__*/_react.default.createElement(LayoutWrapper, {
    Layout: Layout
  }, children)));
};
BaseLayoutProvider.propTypes = {
  /**
   * @deprecated capital letter prop will be remove in future version
   */
  Layout: _propTypes.default.any,
  /**
   * @deprecated capital letter prop will be remove in future version
   */
  Theme: _propTypes.default.any,
  /**
   * Page layout.
   * Can be one of prebuilt layouts (default or side) or can be custom component
   */
  layout: _propTypes.default.any,
  /**
   * Page theme.
   * Can be one of prebuilt themes (default or side) or can be custom object
   */
  theme: _propTypes.default.any,
  /**
   *    The content of the component.
   */
  children: _propTypes.default.node
};
BaseLayoutProvider.defaultProps = {
  Theme: _default2.default,
  // deprecated
  Layout: _default3.default,
  // deprecated
  theme: _default2.default,
  layout: _default3.default
};
var _default = BaseLayoutProvider;
exports.default = _default;
