// harelkit/lib/components/Modal/Head.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Head = _interopRequireDefault(require("./Head.style"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Head = function Head(props) {
  var disableCloseButton = props.disableCloseButton,
      icon = props.icon,
      onClose = props.onClose,
      bo = props.bo;

  var handleCloseButtonClick = function handleCloseButtonClick(event) {
    event.preventDefault();
    onClose(event, 'closeButtonClick');
  };

  var closeButton = !disableCloseButton ? /*#__PURE__*/_react.default.createElement(_Head.default.Close, {
    onClick: handleCloseButtonClick,
    "data-hrl-bo": bo ? "".concat(bo, "-modal-x") : null,
    "aria-label": "\u05E1\u05D2\u05D5\u05E8"
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 15 15"
  }, /*#__PURE__*/_react.default.createElement("g", {
    stroke: "#003C7F",
    strokeWidth: "2",
    fill: "none",
    fillRule: "evenodd",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M2.55 2.55l9.9 9.9M12.45 2.55l-9.9 9.9"
  })))) : null;
  var modalIcon = icon ? /*#__PURE__*/_react.default.createElement(_Head.default.Icon, null, icon) : null;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, modalIcon, closeButton);
};

Head.propTypes = {
  onClose: _propTypes.default.func,
  icon: _propTypes.default.element,
  disableCloseButton: _propTypes.default.bool
};
var _default = Head;
exports.default = _default;
