// common-ui/lib/components/Address/Address.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _index = require("../../index");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _axios = _interopRequireDefault(require("axios"));

var _useReduxField = _interopRequireWildcard(require("../HOC/useReduxField"));

var _misc = require("../../utils/misc");

var _usePrevious = _interopRequireDefault(require("../../utils/usePrevious"));

var _useUpdateEffect = _interopRequireDefault(require("../../utils/useUpdateEffect"));

var _lodash = require("lodash");

var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));

var _Progress = _interopRequireDefault(require("../Progress"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var Address = function Address(props) {
  var _props$childProps2, _props$childProps3, _props$childProps4, _props$childProps5, _props$childProps6, _props$childProps7;

  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var baseUrl = props.baseUrl,
      value = props.value,
      grid = props.grid,
      spacing = props.spacing,
      fieldName = props.name,
      useRedux = props.useRedux,
      proxy = props.proxy,
      hideAppartmentField = props.hideAppartmentField,
      hideMailBoxField = props.hideMailBoxField,
      onDataReceived = props.onDataReceived,
      allowZeroHouseNum = props.allowZeroHouseNum,
      sharedCities = props.sharedCities;
  var dataHrlBo = props['data-hrl-bo'];

  function useMemoState(defState) {
    var _React$useState = _react.default.useState(defState),
        _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
        state = _React$useState2[0],
        setState = _React$useState2[1];

    function smartSetState(newState) {
      if (!(0, _lodash.isEqual)(state, newState)) {
        setState(newState);
      }
    }

    return [state, smartSetState];
  }

  var _useMemoState = useMemoState([]),
      _useMemoState2 = (0, _slicedToArray2.default)(_useMemoState, 2),
      cities = _useMemoState2[0],
      setCities = _useMemoState2[1];

  var _useMemoState3 = useMemoState([]),
      _useMemoState4 = (0, _slicedToArray2.default)(_useMemoState3, 2),
      streets = _useMemoState4[0],
      setStreets = _useMemoState4[1];

  var _useMemoState5 = useMemoState([]),
      _useMemoState6 = (0, _slicedToArray2.default)(_useMemoState5, 2),
      houseNumbers = _useMemoState6[0],
      setHouseNumbers = _useMemoState6[1];

  var _useState = (0, _react.useState)({
    city: false,
    street: false,
    houseNumber: false
  }),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      inProgress = _useState2[0],
      setInProgress = _useState2[1];

  var prevValues = (0, _usePrevious.default)({
    value: value
  });
  var form = useRedux({
    city: value.city,
    street: value.street,
    houseNumber: value.houseNumber,
    postalCode: value.postalCode,
    apartmentNumber: value.apartmentNumber,
    MailBox: value.MailBox
  });
  var getFieldProps = form.getFieldProps,
      handleChange = form.handleChange,
      values = form.values,
      errors = form.errors;
  var city = values.city,
      street = values.street,
      houseNumber = values.houseNumber,
      postalCode = values.postalCode,
      apartmentNumber = values.apartmentNumber,
      MailBox = values.MailBox;

  var Loader = function Loader() {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        position: 'absolute',
        left: '40px',
        top: '32px'
      }
    }, /*#__PURE__*/_react.default.createElement(_Progress.default, {
      size: 18
    }));
  };

  (0, _useUpdateEffect.default)(function () {
    if (prevValues && (0, _lodash.isEqual)(prevValues === null || prevValues === void 0 ? void 0 : prevValues.value, value)) {
      return;
    }

    handleChange('city')((0, _misc.getEventData)(fieldName, value.city));
    handleChange('street')((0, _misc.getEventData)(fieldName, value.street));
    handleChange('postalCode')((0, _misc.getEventData)(fieldName, value.postalCode));
    handleChange('houseNumber')((0, _misc.getEventData)(fieldName, getHouseNumberEventValue(value.houseNumber)));
    handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, value.apartmentNumber));
    handleChange('MailBox')((0, _misc.getEventData)(fieldName, value.MailBox));
  }, [value]);

  var getFullObjectByLabel = function getFullObjectByLabel(object, list) {
    var found = list.find(function (item) {
      return item.label === object.label;
    });
    return found ? found : object;
  };

  (0, _react.useEffect)(function () {
    if (sharedCities) {
      setCities(citiesFormatter(sharedCities));
    } else {
      fetchData('city', pathUrls.cities(), citiesFormatter, setCities);
    }
  }, []);
  (0, _useUpdateEffect.default)(function () {
    props.onCitiesListChange(cities);
  }, [cities]);
  (0, _useUpdateEffect.default)(function () {
    props.onStreetsListChange(streets);
  }, [streets]);
  (0, _useUpdateEffect.default)(function () {
    props.onHouseNumbersListChange(houseNumbers);
  }, [houseNumbers]);
  (0, _useUpdateEffect.default)(function () {
    if (!city) {
      handleChange('street')((0, _misc.getEventData)(fieldName, ''));
      setStreets([]);
    } else if (city.value) {
      fetchData('street', pathUrls.streets(city.value), streetsFormatter, setStreets);
    } else if (city.label) {
      handleChange('city')((0, _misc.getEventData)(fieldName, getFullObjectByLabel(city, cities)));
    }
  }, [city, cities]);
  (0, _useUpdateEffect.default)(function () {
    if (street !== null && street !== void 0 && street.value) {
      fetchData('houseNumber', pathUrls.houseNumbers(city === null || city === void 0 ? void 0 : city.value, street === null || street === void 0 ? void 0 : street.value), postalCodeFormatter, setHouseNumbers);
    } else if (street !== null && street !== void 0 && street.label) {
      handleChange('street')((0, _misc.getEventData)(fieldName, getFullObjectByLabel(street, streets)));
    } else {
      var _props$childProps, _props$childProps$hou;

      var isFreeSolo = (_props$childProps = props.childProps) === null || _props$childProps === void 0 ? void 0 : (_props$childProps$hou = _props$childProps.houseNumber) === null || _props$childProps$hou === void 0 ? void 0 : _props$childProps$hou.freeSolo;

      if (!isFreeSolo) {
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
      }

      setHouseNumbers([]);
    }
  }, [street, streets]);
  (0, _useUpdateEffect.default)(function () {
    var getPostalCode = function getPostalCode() {
      switch (true) {
        case !!(houseNumber && houseNumber.value):
          return houseNumber.value;

        case !!city.postalCode:
          return city.postalCode;

        default:
          var found = cities.find(function (c) {
            return c.value === city.value;
          });
          return (found === null || found === void 0 ? void 0 : found.postalCode) || '';
      }
    };

    if ((0, _misc.isObject)(houseNumber) && !houseNumber.value) setHouseNumbers([]);
    var autoPostalCode = getPostalCode();
    handleChange('postalCode')((0, _misc.getEventData)(fieldName, autoPostalCode));
  }, [houseNumber]);
  (0, _react.useEffect)(function () {
    if (houseNumbers.length && !houseNumber.value && !!houseNumber.label) {
      handleChange('houseNumber')((0, _misc.getEventData)(fieldName, getHouseNumberEventValue(getFullObjectByLabel(houseNumber, houseNumbers))));
    }
  }, [houseNumbers]);
  var urlPrefix = baseUrl || 'https://digital.harel-group.co.il';
  var url = urlPrefix + '/static-data/address';

  if (proxy && proxy !== null) {
    url = proxy + url;
  }

  var pathUrls = {
    cities: function cities() {
      return "".concat(url, "/cities");
    },
    streets: function streets(cityCode) {
      return "".concat(url, "/streets/?cityCode=").concat(cityCode);
    },
    houseNumbers: function houseNumbers(cityCode, streetCode) {
      return "".concat(url, "/postal-codes/?cityCode=").concat(cityCode, "&streetCode=").concat(streetCode, "&nonzeroHouseNum=").concat(allowZeroHouseNum);
    }
  };

  var fetchData = function fetchData(type, url, formatter, setterFunc) {
    setInProgress(function (e) {
      e[type] = true;
      return _objectSpread({}, e);
    });

    _axios.default.get(url).then(function (res) {
      if (!res.data) {
        res.data = [];
      }

      setterFunc(formatter(res.data));
      onDataReceived && onDataReceived({
        type: type,
        data: res.data
      });
    }).catch(function (e) {
      console.log(e);
    }).finally(function () {
      setInProgress(function (e) {
        e[type] = false;
        return _objectSpread({}, e);
      });
    });
  };

  var getHouseNumberEventValue = function getHouseNumberEventValue(houseNumber) {
    return (0, _misc.isObject)(houseNumber) ? {
      label: houseNumber.label,
      value: houseNumber.postalCode || houseNumber.value,
      actualValue: houseNumber.value || houseNumber.actualValue
    } : houseNumber;
  };

  var citiesFormatter = function citiesFormatter(data) {
    if (Array.isArray(data)) {
      return data.map(function (i) {
        return {
          label: trimStr(i.cityName),
          value: i.cityCode,
          postalCode: i.postalCode7 !== 0 ? trimStr(i.postalCode7) : ''
        };
      });
    }

    return [];
  };

  var streetsFormatter = function streetsFormatter(data) {
    if (Array.isArray(data)) {
      return data.map(function (i) {
        return {
          label: trimStr(i.streetName),
          value: i.streetNumber
        };
      });
    }

    return [];
  };

  var postalCodeFormatter = function postalCodeFormatter(data) {
    if (Array.isArray(data)) {
      return data.map(function (i) {
        var entrance = trimStr(i.entrance);
        var houseNumber = trimStr(i.houseNumber);
        return {
          label: entrance ? "".concat(houseNumber, " ").concat(entrance) : houseNumber,
          value: entrance ? "".concat(houseNumber).concat(entrance) : houseNumber,
          postalCode: trimStr(i.postalCode7)
        };
      });
    }

    return [];
  };

  var cityChangeHandler = function cityChangeHandler(event, value, reason) {
    if (reason !== 'clear' && city.value === value.value) {
      return;
    }

    switch (reason) {
      case 'clear':
        handleChange('city')((0, _misc.getEventData)(fieldName, ''));
        handleChange('street')((0, _misc.getEventData)(fieldName, ''));
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('postalCode')((0, _misc.getEventData)(fieldName, ''));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      case 'select-option':
        handleChange('city')((0, _misc.getEventData)(fieldName, value));
        handleChange('street')((0, _misc.getEventData)(fieldName, ''));
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('postalCode')((0, _misc.getEventData)(fieldName, value.postalCode));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      case 'blur':
        handleChange('city')((0, _misc.getEventData)(fieldName, value));
        handleChange('street')((0, _misc.getEventData)(fieldName, ''));
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('postalCode')((0, _misc.getEventData)(fieldName, value.postalCode));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      default:
        handleChange('city')((0, _misc.getEventData)(fieldName, value));
        break;
    }
  };

  var streetChangeHandler = function streetChangeHandler(event, value, reason) {
    switch (reason) {
      case 'clear':
        handleChange('street')((0, _misc.getEventData)(fieldName, ''));
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('postalCode')((0, _misc.getEventData)(fieldName, ''));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      case 'select-option':
        handleChange('street')((0, _misc.getEventData)(fieldName, value));
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      case 'blur':
        // select via BLUR
        var isFreeSoloResult = typeof value === 'string';

        if (isFreeSoloResult && street.label !== value) {
          handleChange('street')((0, _misc.getEventData)(fieldName, {
            label: value,
            value: null
          }));
        } // select via TAB


        if (value !== null && value !== void 0 && value.value) {
          handleChange('street')((0, _misc.getEventData)(fieldName, value));
          handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
          handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
          handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        }

        break;

      default:
        handleChange('street')((0, _misc.getEventData)(fieldName, value));
        break;
    }
  };

  var houseNumberChangeHandler = function houseNumberChangeHandler(event, value, reason) {
    switch (reason) {
      case 'clear':
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('postalCode')((0, _misc.getEventData)(fieldName, ''));
        handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, ''));
        handleChange('MailBox')((0, _misc.getEventData)(fieldName, ''));
        break;

      case 'blur':
        // select via BLUR
        var isFreeSoloResult = typeof value === 'string';

        if (isFreeSoloResult && houseNumber.label !== value) {
          handleChange('houseNumber')((0, _misc.getEventData)(fieldName, {
            label: value,
            value: null
          }));
        } // select via TAB


        if (value !== null && value !== void 0 && value.value) {
          handleChange('houseNumber')((0, _misc.getEventData)(fieldName, getHouseNumberEventValue(value)));
        }

        break;

      default:
        handleChange('houseNumber')((0, _misc.getEventData)(fieldName, getHouseNumberEventValue(value)));
        break;
    }
  };

  var trimStr = function trimStr(value) {
    return value || value === 0 ? value.toString().trim() : '';
  };

  (0, _react.useEffect)(function () {
    if (!inProgress.city && !inProgress.street && !inProgress.houseNumber) {
      (props === null || props === void 0 ? void 0 : props.onLoaded) && props.onLoaded();
    }
  }, [inProgress === null || inProgress === void 0 ? void 0 : inProgress.city, inProgress === null || inProgress === void 0 ? void 0 : inProgress.street, inProgress === null || inProgress === void 0 ? void 0 : inProgress.houseNumber]);
  return /*#__PURE__*/_react.default.createElement(_index.Grid, {
    container: true,
    spacing: spacing,
    id: props.id,
    "data-hrl-bo": "".concat(dataHrlBo, "-mainBox")
  }, /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.city), /*#__PURE__*/_react.default.createElement(_index.Autocomplete, (0, _extends2.default)({
    label: "\u05E2\u05D9\u05E8 / \u05D9\u05E9\u05D5\u05D1"
  }, getFieldProps('city'), {
    options: cities,
    defaultValue: '',
    displayEmpty: true,
    value: city,
    getOptionSelected: function getOptionSelected(option, value) {
      var _value$value;

      return option.value.toString() === ((_value$value = value.value) === null || _value$value === void 0 ? void 0 : _value$value.toString());
    },
    getOptionLabel: function getOptionLabel(option) {
      return option.label;
    },
    onChange: cityChangeHandler,
    "data-hrl-bo": "".concat(dataHrlBo, "-city"),
    "data-hrl-is-in-progress": inProgress.city,
    autoHighlight: true,
    autoSelect: true,
    inputProps: props.inputProps,
    role: props.role,
    InputProps: props.InputProps,
    isMatchFromStart: props.isFilterByStartWith,
    disablePortal: true
  }, (_props$childProps2 = props.childProps) === null || _props$childProps2 === void 0 ? void 0 : _props$childProps2.city)), inProgress.city && /*#__PURE__*/_react.default.createElement(Loader, null)), /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.street), /*#__PURE__*/_react.default.createElement(_index.Autocomplete, (0, _extends2.default)({
    label: "\u05E8\u05D7\u05D5\u05D1"
  }, getFieldProps('street'), {
    options: streets,
    value: street,
    getOptionSelected: function getOptionSelected(option, value) {
      var _value$value2;

      return option.value.toString() === ((_value$value2 = value.value) === null || _value$value2 === void 0 ? void 0 : _value$value2.toString());
    },
    defaultValue: '',
    displayEmpty: true,
    onChange: streetChangeHandler,
    "data-hrl-bo": "".concat(dataHrlBo, "-street"),
    "data-hrl-is-in-progress": inProgress.street,
    autoHighlight: true,
    autoSelect: true,
    inputProps: props.inputProps,
    role: props.role,
    InputProps: props.InputProps,
    isMatchFromStart: props.isFilterByStartWith,
    disablePortal: true
  }, (_props$childProps3 = props.childProps) === null || _props$childProps3 === void 0 ? void 0 : _props$childProps3.street)), inProgress.street && /*#__PURE__*/_react.default.createElement(Loader, null)), /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.houseNumber), /*#__PURE__*/_react.default.createElement(_index.Autocomplete, (0, _extends2.default)({
    label: "\u05DE\u05E1\u05E4\u05E8 \u05D1\u05D9\u05EA"
  }, getFieldProps('houseNumber'), {
    options: houseNumbers,
    value: houseNumber,
    getOptionSelected: function getOptionSelected(option, value) {
      var _value$value3;

      return option.value.toString() === ((_value$value3 = value.value) === null || _value$value3 === void 0 ? void 0 : _value$value3.toString()) || option.postalCode === (value === null || value === void 0 ? void 0 : value.value);
    },
    defaultValue: '',
    displayEmpty: true,
    onChange: houseNumberChangeHandler,
    "data-hrl-bo": "".concat(dataHrlBo, "-houseNumber"),
    "data-hrl-is-in-progress": inProgress.houseNumber,
    autoHighlight: true,
    autoSelect: true,
    inputProps: props.inputProps,
    role: props.role,
    InputProps: props.InputProps,
    disablePortal: true
  }, (_props$childProps4 = props.childProps) === null || _props$childProps4 === void 0 ? void 0 : _props$childProps4.houseNumber)), inProgress.houseNumber && /*#__PURE__*/_react.default.createElement(Loader, null)), !hideAppartmentField && /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.apartmentNumber), /*#__PURE__*/_react.default.createElement(_index.TextField, (0, _extends2.default)({
    label: "\u05D3\u05D9\u05E8\u05D4"
  }, getFieldProps('apartmentNumber'), {
    value: apartmentNumber,
    maxLength: 4,
    type: "phone",
    "data-hrl-bo": "".concat(dataHrlBo, "-apartmentNumber"),
    onChange: function onChange(e) {
      return handleChange('apartmentNumber')((0, _misc.getEventData)(fieldName, e.target.value));
    }
  }, (_props$childProps5 = props.childProps) === null || _props$childProps5 === void 0 ? void 0 : _props$childProps5.apartmentNumber))), !hideMailBoxField && /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.MailBox), /*#__PURE__*/_react.default.createElement(_index.TextField, (0, _extends2.default)({
    label: "\u05EA\u05D9\u05D1\u05EA \u05D3\u05D5\u05D0\u05E8"
  }, getFieldProps('MailBox'), {
    value: MailBox || '',
    maxLength: 5,
    mask: '99999',
    type: "tel",
    "data-hrl-bo": "".concat(dataHrlBo, "-MailBox"),
    onChange: function onChange(e) {
      return handleChange('MailBox')((0, _misc.getEventData)(fieldName, e.target.value));
    }
  }, (_props$childProps6 = props.childProps) === null || _props$childProps6 === void 0 ? void 0 : _props$childProps6.MailBox))), /*#__PURE__*/_react.default.createElement(_index.Grid, (0, _extends2.default)({
    item: true,
    xs: 12
  }, grid.postalCode), /*#__PURE__*/_react.default.createElement(_index.TextField, (0, _extends2.default)({
    label: "\u05DE\u05D9\u05E7\u05D5\u05D3"
  }, getFieldProps('postalCode'), {
    value: postalCode,
    mask: '9999999',
    maxLength: 7,
    "data-hrl-bo": "".concat(dataHrlBo, "-postalCode"),
    onChange: function onChange(e) {
      return handleChange('postalCode')((0, _misc.getEventData)(fieldName, e.target.value));
    }
  }, (_props$childProps7 = props.childProps) === null || _props$childProps7 === void 0 ? void 0 : _props$childProps7.postalCode))));
};

var singleObjectValueStruct = _propTypes.default.shape({
  label: _propTypes.default.any.isRequired,
  value: _propTypes.default.any.isRequired
});

var sizesStruct = _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]);

var singleObjectSizeStruct = _propTypes.default.shape({
  xs: sizesStruct,
  md: sizesStruct,
  lg: sizesStruct,
  xl: sizesStruct
});

var propTypes = {
  // === Component Settings === //

  /** Static Data base URL */
  baseUrl: _propTypes.default.string,

  /** The value of the input element, required for a controlled component. */
  value: _propTypes.default.shape({
    city: singleObjectValueStruct,
    street: singleObjectValueStruct,
    houseNumber: singleObjectValueStruct,
    postalCode: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
    apartmentNumber: _propTypes.default.string,
    MailBox: _propTypes.default.string
  }),

  /** Name attribute of the input element. */
  name: _propTypes.default.string,

  /** hide Appartment Field */
  hideAppartmentField: _propTypes.default.bool,
  hideMailBoxField: _propTypes.default.bool,

  /** Custom child props */
  childProps: _propTypes.default.shape({
    city: _propTypes.default.object,
    street: _propTypes.default.object,
    houseNumber: _propTypes.default.object,
    postalCode: _propTypes.default.object,
    singleObjectSizeStruct: _propTypes.default.object
  }),

  /** Allow to get house number starting from 0 */
  allowZeroHouseNum: _propTypes.default.bool,

  /** Data hrl bo attribute for automation testing. */
  'data-hrl-bo': _propTypes.default.string,

  /** value moved to isMatchFromStart from material */
  isFilterByStartWith: _propTypes.default.bool,

  /** When you use multiple address components on page you can */
  sharedCities: _propTypes.default.array,

  /** Show loading static data indication */
  showLoader: _propTypes.default.bool,
  // === Grid Settings === //

  /** The grid items sizing */
  grid: _propTypes.default.shape({
    city: singleObjectSizeStruct,
    street: singleObjectSizeStruct,
    houseNumber: singleObjectSizeStruct,
    postalCode: singleObjectSizeStruct,
    apartmentNumber: singleObjectSizeStruct,
    MailBox: singleObjectSizeStruct
  }),

  /** The grid container spacing */
  spacing: _propTypes.default.number,
  // === Callbacks === //

  /** Callback fired when value is changed */
  onChange: _propTypes.default.func,

  /** Callback fired when loaded cities */
  onCitiesListChange: _propTypes.default.func,

  /** Callback fired when loaded streets */
  onStreetsListChange: _propTypes.default.func,

  /** Callback fired when loaded house numbers */
  onHouseNumbersListChange: _propTypes.default.func,

  /** Callback fired when finish any request to server */
  onDataReceived: _propTypes.default.func,

  /**
   * @deprecated
   * @hidden
   * Callback fired when value is changed
   * */
  onBlur: _propTypes.default.func,
  // === Dev Props === //

  /**
   * @hidden
   * Pass proxy server url to allow on local
   * */
  proxy: _propTypes.default.string,

  /**
   * Add Custom ROLE to main div of Autocomplate, default value is combobox, role=null remove the attribute
   **/
  role: _propTypes.default.string,

  /**
   * Add Custom attributes to input element in Autocomplate, default value is null
   **/
  inputProps: _propTypes.default.object,

  /**
   * Add Custom attributes to wrapper (DIV) of input element in Autocomplate, default value is null
   **/
  InputProps: _propTypes.default.object,

  /**
   * when the component loaded you can run callback function
   **/
  onLoaded: _propTypes.default.func,

  /**
   * ID of component, default is null
   **/
  id: _propTypes.default.string
};
var defaultProps = {
  baseUrl: 'https://digital.harel-group.co.il',
  onDataReceived: function onDataReceived() {},
  onChange: function onChange(e) {},
  spacing: 2,
  allowZeroHouseNum: false,
  grid: {
    city: {
      xs: 6
    },
    street: {
      xs: 6
    },
    houseNumber: {
      xs: 4
    },
    postalCode: {
      xs: 4
    },
    apartmentNumber: {
      xs: 4
    },
    MailBox: {
      xs: 4
    }
  },
  'data-hrl-bo': 'address',
  value: {
    city: '',
    street: '',
    houseNumber: '',
    postalCode: '',
    apartmentNumber: '',
    MailBox: ''
  },
  name: 'address',
  proxy: null,
  onCitiesListChange: function onCitiesListChange() {},
  onStreetsListChange: function onStreetsListChange() {},
  onHouseNumbersListChange: function onHouseNumbersListChange() {},
  hideAppartmentField: true,
  hideMailBoxField: true,
  isFilterByStartWith: false,
  showLoader: true,
  role: 'combobox',
  inputProps: null,
  InputProps: null,
  id: null
};
Address.propTypes = _objectSpread(_objectSpread({}, propTypes), _useReduxField.propTypes);
Address.defaultProps = _objectSpread({}, defaultProps);

var AddressInner = function AddressInner(props) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, (0, _useReduxField.default)(Address)(props));
};

AddressInner.propTypes = _objectSpread(_objectSpread({}, propTypes), _useReduxField.propTypes);
AddressInner.defaultProps = _objectSpread({}, defaultProps);
AddressInner.displayName = 'Address';
var _default = AddressInner;
exports.default = _default;
