// common-ui/lib/components/Toggle/Toggle.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: 'relative',
      borderRadius: theme.spacing(2),
      lineHeight: '1',
      display: 'flex',
      marginTop: function marginTop(props) {
        return props.label ? '20px' : 0;
      },
      width: function width(props) {
        return props.asSwitch ? '96px' : null;
      },
      '& > label': {
        position: 'relative',
        height: function height(props) {
          return props.labelHeight;
        },
        width: function width(props) {
          return props.labelWidth;
        }
      },
      '& > input': {
        width: '0px',
        margin: 0,
        height: 0
      },
      '& > label:not(:last-child)': {
        borderLeft: '1px solid rgba(101, 101, 101, 0.1)'
      },
      '& > input:disabled + label:not(:last-child)': {
        borderLeft: '1px solid rgba(101, 101, 101, 0.1)'
      },
      '&.isError > label': {
        border: '1px solid red',
        ' &:not(:last-child)': {
          borderLeft: '1px solid rgba(101, 101, 101, 0.1)'
        }
      }
    }
  };
});
var _default = useStyles;
exports.default = _default;
