// common-ui/lib/components/BaseLayoutProvider/Themes/base/base.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _core = require("@material-ui/core");
var _input;
var spacing = 8;
var direction = 'rtl';
var shape = {};

/**
 * Palette
 */
var WHITE = '#FFFFFF';
var BLACK = '#000000';
var palette = {
  primary: {
    light: '#6099dc',
    main: '#2d76cb',
    dark: '#2769b4',
    contrastText: WHITE,
    text: WHITE,
    lightText: WHITE
  },
  secondary: {
    superLight: '#e4e9f2',
    light: '#339FE5',
    main: '#006cb2',
    dark: '#005399',
    text: WHITE,
    contrastText: WHITE,
    lightText: BLACK
  },
  error: {
    superLight: '#ffe5db',
    light: '#F54437',
    main: '#DB2A1D',
    dark: '#C21104',
    contrastText: WHITE,
    text: WHITE,
    lightText: BLACK
  },
  warn: {
    light: '#FFD442',
    main: '#FBBA28',
    dark: '#E2A10F',
    contrastText: BLACK,
    text: BLACK,
    lightText: BLACK
  },
  info: {
    light: '#6099dc',
    main: '#2d76cb',
    dark: '#2769b4',
    contrastText: WHITE,
    text: WHITE,
    lightText: BLACK
  },
  success: {
    light: '#67BC57',
    main: '#4DA23D',
    dark: '#348924',
    contrastText: BLACK,
    text: BLACK,
    lightText: BLACK
  },
  background: {
    default: WHITE
  },
  text: {
    primary: '#003c7f',
    secondary: '#2d76cb',
    disabled: '#5c6266',
    hint: 'rgba(0, 63, 128, 0.38)'
  }
};

/**
 * Typography
 */
var typography = {
  fontFamily: '"mfw_protocolharel", Arial',
  fontSize: 15,
  htmlFontSize: 15
};

/**
 * Breakpoints
 */
var breakpoints = {
  keys: ['xs', 'sm', 'md', 'lg', 'xl'],
  values: {
    xs: 0,
    sm: 518,
    md: 765,
    lg: 1013,
    xl: 1440
  }
};

/**
 * Overrides
 */
var overrides = {
  MuiIconButton: {
    root: {
      padding: "".concat(spacing, "px"),
      '&[class*="Mui-focusVisible"]': {
        backgroundColor: 'rgba(0, 108, 178, 0.32)'
      }
    }
  },
  MuiContainer: {
    maxWidthXl: {
      '@media(minWidth: 1400px)': {
        width: '1400px'
      }
    }
  },
  MuiToggle: {
    margin: 0
  },
  MuiSnackbarContent: {
    root: {
      backgroundColor: '#fff',
      color: palette.primary.main
    },
    action: {
      marginRight: 'auto',
      marginLeft: "-".concat(spacing, "px"),
      paddingRight: "".concat(spacing * 2, "px"),
      paddingLeft: null
    }
  },
  MuiBadge: {
    badge: {
      color: '#ffffff !important'
    }
  },
  MuiDialog: {
    paper: {
      overflow: 'hidden'
    }
  },
  MuiPaper: {
    rounded: {
      borderRadius: '10px'
    }
  },
  MuiDialogActions: {
    root: {
      padding: "".concat(spacing * 2, "px ").concat(spacing * 3, "px")
    }
  },
  MuiButtonBase: {
    root: {
      border: '1px solid transparent',
      '&$disabled': {
        '& > span': {
          color: '#5c6266'
        }
      }
    }
  },
  MuiRadio: {
    colorPrimary: {
      color: "".concat(palette.error.main, " !important")
    }
  },
  MuiRangeDatepicker: {
    dayTextColor: '#2d76cb',
    highlightDayTextColor: palette.primary.main,
    hoveredDayTextColor: palette.primary.main,
    trailColor: '#e7f4e0',
    trailHoverColor: '#e7f4e0',
    fromToColor: '#cfeac2'
  },
  MuiListItem: {
    root: {
      textAlign: 'right'
    }
  },
  MuiLink: {
    underlineHover: {
      '&:hover': {
        textDecorationSkipInk: 'none'
      }
    },
    underlineAlways: {
      textDecorationSkipInk: 'none'
    }
  },
  MuiButton: {
    label: {
      lineHeight: 1
    },
    root: {
      boxShadow: 'none',
      border: '3px solid rgba(255,255,255,0)',
      '&:not(:last-child)': {
        marginLeft: '16px'
      },
      '&:focus': {
        boxShadow: '0 0 5 rgba(0,0,0,0.3) !important',
        border: '1px solid rgba(255,255,255,0.5)'
      },
      '&:focus::before': {
        borderRadius: '24px',
        bottom: '-4px',
        content: '\'\'',
        display: 'block',
        left: '-4px',
        position: 'absolute',
        right: '-4px',
        top: '-4px',
        border: '1px solid #2d76cb'
      },
      padding: '6px 16px',
      height: '36px',
      color: '#2d76cb',
      fontSize: '16px',
      borderRadius: '24px',
      fontWeight: 'bold',
      textTransform: 'initial',
      '&:hover': {
        boxShadow: '0 0 1px 1px #2d76cb !important',
        border: '3px solid #fff'
      }
    },
    contained: {
      boxShadow: 'none',
      '&:active': {
        boxShadow: 'none'
      },
      '&$focusVisible': {
        boxShadow: 'none'
      },
      color: 'white',
      '&[class*="Mui-disabled"]': {
        boxShadow: '0px 2px 4px -1px rgba(0,0,0,0.2)',
        backgroundColor: '#e2ecf9'
      }
    },
    containedSizeSmall: {
      height: '28.8px',
      fontSize: '14px'
    },
    containedSizeLarge: {
      height: '43.2px',
      fontSize: '20px'
    },
    outlined: {
      background: '#fff',
      padding: '6px 16px',
      borderRadius: '24px',
      height: '36px',
      color: "".concat(palette.primary.main, " !important"),
      backgroundColor: 'white',
      border: "1px solid ".concat(palette.primary.main, " !important"),
      '&:focus': {
        backgroundColor: "rgba(0, 36, 71, 0.04) !important"
      },
      '&.Mui-disabled': {
        borderColor: '#5c6266'
      }
    },
    outlinedSizeSmall: {
      height: '28.8px',
      fontSize: '14px'
    },
    outlinedSizeLarge: {
      height: '43.2px',
      fontSize: '20px'
    },
    outlinedPrimary: {
      '&:focus': {
        border: '1px solid rgba(0, 63, 128, 0.5)'
      }
    },
    outlinedSecondary: {
      '&:focus': {
        border: '1px solid rgba(0, 108, 178, 0.5)'
      }
    },
    containedPrimary: {
      backgroundColor: '#2d76cb',
      '&:hover': {
        backgroundColor: '#4387d5'
      },
      '&:focus': {
        backgroundColor: '#2d76cb'
      },
      '&:active': {
        backgroundColor: '#2561a7'
      }
    },
    text: {
      padding: '6px 16px'
    },
    textSizeSmall: {
      height: '28.8px',
      fontSize: '14px'
    },
    textSizeLarge: {
      height: '43.2px',
      fontSize: '20px'
    }
  },
  MuiInputLabel: {
    root: {
      transformOrigin: 'top right',
      pointerEvents: 'none',
      display: 'block'
    },
    formControl: {
      color: '#003f80',
      paddingRight: '7px',
      paddingTop: '7px',
      top: '0',
      left: 'auto',
      right: '4px',
      position: 'absolute',
      transform: 'translate(0, 13px)',
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      width: '100%'
    },
    shrink: {
      transformOrigin: 'top right',
      transform: 'translate(0, 1.5px) scale(0.8)',
      width: '125%',
      fontWeight: '300',
      paddingTop: '7px'
    },
    '&[class*="Mui-focused"]': {
      fontWeight: 100
    }
  },
  MuiTooltip: {
    arrow: {
      '&[x-placement*="left"] $arrow': {
        /* tricky doubly-quoted empty string so mui parses it as truly empty */
        content: '""',
        display: 'block',
        width: '0',
        height: '0',
        color: 'red',
        position: 'absolute',
        borderLeft: '10px solid transparent',
        borderRight: '10px solid transparent',
        /* border color should probably match whatever your tooltip color is */
        borderTop: '10px solid #000',
        left: 'calc(50% - 10px)',
        top: '100%'
      }
    }
  },
  MuiTextField: {
    root: {
      width: '100%',
      borderRadius: '4px',
      '& input:-webkit-autofill': {
        '-webkit-box-shadow': 'none',
        '-webkit-text-fill-color': palette.text.main
      }
    }
  },
  MuiFormHelperText: {
    root: {
      textAlign: 'right',
      marginTop: 2,
      color: '#5c6266',
      paddingRight: 10
    }
  },
  MuiInputBase: {
    root: {
      '&[class*="Mui-disabled"]': {
        color: '#777777'
      },
      background: 'inherit',
      '& > input': {
        borderColor: 'transparent'
      }
    },
    inputAdornedStart: {
      marginRight: '8px'
    },
    input: (_input = {
      color: '#003c7f',
      fontSize: '20px',
      lineHeight: '20px',
      height: '1.25em',
      paddingRight: '9px !important'
    }, (0, _defineProperty2.default)(_input, "@media (max-width: ".concat(breakpoints.values.sm, "px)"), {
      paddingRight: '10px !important'
    }), (0, _defineProperty2.default)(_input, '&:focus', {
      outlineWidth: '0px !important'
    }), _input)
  },
  MuiChip: {
    deleteIcon: {
      margin: '0 -5px 0 6px'
    },
    root: {
      minHeight: '32px',
      height: 'auto '
    },
    label: {
      whiteSpace: 'normal'
    }
  },
  MuiInput: {
    input: {
      padding: '6px 4px 5px 4px',
      '&::placeholder': {
        fontWeight: 'bolder',
        color: '#012752',
        fontSize: '16px',
        lineHeight: '20px'
      },
      '&::-webkit-outer-spin-button': {
        '-webkit-appearance': 'none',
        margin: 0
      },
      '&::-webkit-inner-spin-button': {
        '-webkit-appearance': 'none',
        margin: 0
      },
      '&:-Webkit-autofill': {
        boxShadow: 'none'
      },
      '&:-Webkit-autofill:hover': {
        boxShadow: 'none'
      }
    },
    formControl: {
      fontSize: '1rem'
    },
    underline: {
      '&[class*="Mui-disabled"]:before': {
        'border-bottom-style': 'hidden',
        borderBottomColor: 'rgba(0, 0, 0, 0.42)'
      },
      '&:before': {
        borderBottom: '1px solid #003c7f'
      },
      '&:after': {
        borderBottomColor: '#2769b4'
      },
      '&:hover:not($disabled):not($error):before': {
        borderBottomColor: '#2769b4'
      }
    }
  },
  MuiMenuItem: {
    root: {
      fontSize: '1rem',
      minHeight: null,
      lineHeight: 1.5,
      color: '#003c7f'
    }
  },
  MuiList: {
    root: {
      direction: 'rtl'
    },
    padding: {
      paddingTop: '4px',
      paddingBottom: '4px'
    }
  },
  MuiTabs: {
    root: {
      marginTop: spacing * 3,
      marginBottom: spacing * 3
    },
    scroller: {
      '&:after': {
        content: '""',
        display: 'block',
        width: '100%',
        borderBottom: '3px solid #E3EBF6'
      }
    },
    indicator: {
      borderBottom: '3px solid #0088C9'
    },
    vertical: {
      borderLeft: '1px solid #E3EBF6',
      '& [class*="MuiButtonBase-root"]': {
        padding: '10px 16px'
      },
      '& [class*="MuiTabs-indicator"]': {
        right: 'auto',
        left: 0
      },
      '& [class*="MuiTabs-scroller"]': {
        '&:after': {
          content: '""',
          display: 'none',
          width: '100%',
          borderBottom: '0px solid #E3EBF6'
        }
      },
      '& [class*="MuiTabs-scrollButtons"]': {
        '& svg': {
          transform: 'rotate(-90deg)'
        }
      }
    }
  },
  MuiTab: {
    root: {
      minHeight: null,
      padding: '20px 16px',
      fontSize: '1.1rem',
      '&:focus': {
        backgroundColor: 'rgba(0, 63, 128, 0.08)'
      }
    }
  },
  MuiSelect: {
    root: {
      backgroundColor: '#e2ecf9',
      borderRadius: '4px',
      paddingRight: '11px !important'
    },
    select: {
      paddingLeft: '32px',
      paddingTop: '24px',
      backgroundColor: 'none',
      borderRadius: '4px 4px 0 0',
      '&:focus': {
        //borderRadius: spacing,
        backgroundColor: 'none'
      },
      '&[class*="MuiSelect-selectMenu"]': {
        paddingRight: '8px !important',
        paddingLeft: '32px',
        minHeight: '23px'
      }
    },
    icon: {
      color: "#003c7f !important",
      width: '20px !important',
      height: '20px !important',
      top: '21px !important',
      left: '8px !important',
      right: 'auto !important'
    },
    iconFilled: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    }
  },
  MuiNativeSelect: {
    root: {
      width: '100%'
    },
    select: {
      paddingLeft: '32px',
      paddingRight: '8px !important',
      backgroundColor: 'none',
      borderRadius: "".concat(spacing, "px ").concat(spacing, "px 0 0"),
      '&:focus': {
        borderRadius: spacing,
        backgroundColor: 'none'
      },
      '&[class*="MuiSelect-selectMenu"]': {
        paddingRight: '4px !important'
      }
    },
    icon: {
      left: '8px',
      right: 'auto',
      top: 'calc(50% - 13px)',
      color: "#003c7f !important",
      width: '20px !important',
      height: '20px !important'
    },
    iconFilled: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    }
  },
  MuiFormLabel: {
    root: {
      color: palette.text.primary,
      fontSize: '1.25rem',
      zIndex: 2,
      padding: 0,
      paddingLeft: 4,
      fontWeight: '500',
      '&[class*="Mui-disabled"]': {
        color: '#5c6266'
      }
    },
    asterisk: {
      color: '#D32704',
      letterSpacing: -2
    }
  },
  MuiCheckbox: {
    root: {
      color: '#DDE4ED',
      padding: '0 !important',
      marginLeft: '5px !important'
    },
    colorPrimary: {
      color: "".concat(palette.error.main, " !important")
    },
    colorSecondary: {
      color: '#2D76CB',
      '&[class*="Mui-disabled"]': {
        color: '#5c6266'
      }
    }
  },
  MuiExpansionPanelSummary: {
    root: {
      padding: '0 16px 0 0',
      background: '#E3EBF6',
      '&$expanded': {
        minHeight: '48px'
      },
      content: {
        '&$expanded': {
          margin: '12px 0'
        }
      }
    }
  },
  MuiSvgIcon: {
    colorAction: {
      color: '#0088C9'
    }
  },
  MuiExpansionPanelDetails: {
    root: {
      padding: '16px'
    }
  },
  MuiSwitch: {
    input: {
      left: null,
      width: null
    }
  },
  MuiFormControl: {
    root: {
      width: '100%',
      '&[class*="MuiFormControl-CheckboxSelect"]': {
        '& label': {
          paddingTop: '7px'
        }
      },
      '&[class*="MuiFormControl-CheckboxSelect"] > label + [class*="MuiInput-formControl"]': {
        marginTop: '0'
      },
      '&[class*="MuiFormControl-Select"]': {
        '& label': {
          paddingTop: '7px'
        }
      },
      '&[class*="MuiFormControl-Select"] > label + [class*="MuiInput-formControl"]': {
        marginTop: '0'
      },
      '&[class*="MuiTextField-root"] > label + [class*="MuiInput-formControl"]': {
        marginTop: '0',
        paddingTop: '16px',
        background: '#e2ecf9',
        borderRadius: '4px'
      },
      '&[class*="MuiTextField-root"] > [class*="MuiInputBase-multiline"]': {
        padding: '22px 0 7px !important'
      }
    }
  },
  MuiFormControlLabel: {
    root: {
      marginRight: '-9px',
      marginLeft: '0px'
    },
    labelPlacementStart: {
      marginLeft: null,
      marginRight: null
    },
    label: {
      fontSize: '1.2rem',
      fontWeight: 500,
      '&[class*="Mui-disabled"]': {
        color: '#5c6266'
      }
    }
  },
  MuiInputAdornment: {
    positionStart: {
      marginRight: '3.5px'
    }
  },
  MuiDivider: {
    root: {
      backgroundColor: '#E3EBF6',
      height: '3px',
      borderRadius: '4px'
    }
  },
  MuiTypography: {
    h1: (0, _defineProperty2.default)({
      fontSize: '32px',
      lineHeight: '36px',
      fontWeight: 500,
      color: palette.text.primary
    }, "@media (max-width: ".concat(breakpoints.values.sm, "px)"), {
      fontSize: '24px',
      lineHeight: '27px'
    }),
    h2: (0, _defineProperty2.default)({
      fontSize: '32px',
      lineHeight: '36px',
      fontWeight: 500,
      color: palette.text.primary
    }, "@media (max-width: ".concat(breakpoints.values.sm, "px)"), {
      fontSize: '24px',
      lineHeight: '27px'
    }),
    h3: (0, _defineProperty2.default)({
      fontSize: '24px',
      lineHeight: '30px'
    }, "@media (max-width: ".concat(breakpoints.values.sm, "px)"), {
      fontSize: '20px',
      lineHeight: '25px'
    }),
    h4: {
      fontSize: '20px'
    },
    h5: {
      fontSize: '16px',
      lineHeight: '25px'
    },
    h6: {
      fontSize: '13px'
    },
    body1: {
      fontSize: '16px',
      lineHeight: '20px'
    },
    body2: {
      fontSize: '14px',
      lineHeight: '20px'
    }
  },
  MuiAutocomplete: {
    clearIndicator: {
      padding: '1px !important',
      '& [class*="MuiSvgIcon-root"]': {
        fill: '#003c7f'
      }
    },
    popupIndicator: {
      padding: '0px !important',
      border: '0px',
      '& [class*="MuiSvgIcon-root"]': {
        color: '#003c7f'
      }
    },
    inputRoot: {
      paddingRight: '6px !important',
      paddingLeft: '32px',
      '&[class*="MuiInput-root"]': {
        paddingRight: 0
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child': {
        padding: '6.5px 3.5px 3.5px 3.5px !important',
        paddingRight: 0
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]': {
        borderColor: 'transparent'
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child::-ms-clear': {
        display: 'none',
        width: 0,
        height: 0
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child::-ms-reveal': {
        display: 'none',
        width: 0,
        height: 0
      },
      '&[class*="MuiFilledInput-adornedEnd"] [class*="MuiAutocomplete-endAdornment"]': {
        left: 7,
        right: 'auto !important'
      }
    },
    endAdornment: {
      textAlign: 'left',
      right: 'auto',
      color: "#003f80 !important",
      top: '5px !important',
      left: '3px !important',
      width: '50px !important',
      height: '20px !important'
    }
  },
  MuiPickersYear: {
    root: {
      '&:focus': {
        border: '1px solid ' + _core.colors.grey[400]
      }
    }
  },
  MuiPickersMonth: {
    root: {
      '&:focus': {
        border: '1px solid ' + _core.colors.grey[400]
      }
    }
  },
  MuiCircularProgress: {
    root: {
      '-ms-overflow-x': 'hidden',
      '-ms-overflow-y': 'hidden'
    }
  },
  MuiPickersDay: {
    dayDisabled: {
      color: 'rgba(0, 63, 128, 0.38) !important',
      pointerEvents: 'none'
    }
  },
  MuiGrid: {
    item: {
      position: 'relative'
    }
  },
  MuiPickersCalendarHeader: {
    dayLabel: {
      color: 'rgba(0, 0, 0, 0.54)',
      fontWeight: 'bold'
    }
  },
  MuiPickersToolbarText: {
    toolbarTxt: {
      color: '#FFFFFF'
    }
  }
};
var _default = {
  palette: palette,
  spacing: spacing,
  typography: typography,
  direction: direction,
  breakpoints: breakpoints,
  overrides: overrides,
  shape: shape
};
exports.default = _default;
