// common-ui/lib/components/BaseLayoutProvider/Layouts/side/side.style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _sun = _interopRequireDefault(require("../images/sun.svg"));

var _balloon = _interopRequireDefault(require("../images/balloon1.svg"));

var _balloon2 = _interopRequireDefault(require("../images/balloon2.svg"));

var _hill = _interopRequireDefault(require("../images/hill.svg"));

var _town = _interopRequireDefault(require("../images/town.svg"));

var _road = _interopRequireDefault(require("../images/road.svg"));

var _cloud = _interopRequireDefault(require("../images/cloud1.svg"));

var _cloud2 = _interopRequireDefault(require("../images/cloud2.svg"));

var _cloud3 = _interopRequireDefault(require("../images/cloud3.svg"));

var _cloud4 = _interopRequireDefault(require("../images/cloud4.svg"));

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _main;

  return {
    '@keyframes float-balloon-1': {
      '0%': {
        transform: 'translate(0,0)'
      },
      '50%': {
        transform: 'translate(0,10px)'
      },
      '100%': {
        transform: 'translate(0,0)'
      }
    },
    '@keyframes float-balloon-2': {
      '0%': {
        transform: 'translate(0,0)'
      },
      '50%': {
        transform: 'translate(2px,4px)'
      },
      '100%': {
        transform: 'translate(0,0)'
      }
    },
    '@keyframes cloud-4': {
      '0%': {
        transform: 'translate(314px,0)'
      },
      '100%': {
        transform: 'translate(814px, 0)'
      }
    },
    '@keyframes cloud-3': {
      '0%': {
        transform: 'translate(417px,0)'
      },
      '100%': {
        transform: 'translate(817px, 0)'
      }
    },
    '@keyframes move-decor': {
      from: {
        backgroundPosition: '0 0'
      },
      to: {
        backgroundPosition: '10000px 0'
      }
    },
    root: {
      background: 'linear-gradient(to bottom,#edebe2 0,#f4f2ee 38%,#fff 100%)',
      display: 'flex'
    },
    main: (_main = {
      zIndex: 10,
      flexGrow: 1,
      background: '#ffffff',
      boxShadow: '-1px 0 8px 1px rgba(204,204,204,.4)'
    }, (0, _defineProperty2.default)(_main, theme.breakpoints.down('sm'), {
      maxWidth: '100%'
    }), (0, _defineProperty2.default)(_main, theme.breakpoints.up('md'), {
      maxWidth: '60%'
    }), (0, _defineProperty2.default)(_main, theme.breakpoints.up('lg'), {
      maxWidth: '40%'
    }), _main),
    container: {
      paddingTop: theme.spacing(3),
      paddingBottom: theme.spacing(3),
      minHeight: '100vh'
    },
    images: {
      width: '100%',
      overflow: 'hidden',
      position: 'fixed',
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    sun: {
      position: 'absolute',
      height: '80px',
      width: '80px',
      top: '30px',
      left: '50%',
      zIndex: 1,
      backgroundImage: 'url(' + _sun.default + ')'
    },
    road: {
      position: 'absolute',
      left: '0',
      right: '0',
      bottom: '-7px',
      zIndex: '1',
      backgroundImage: 'url(' + _road.default + ')',
      display: 'block',
      width: '100000%',
      backgroundRepeat: 'repeat',
      backgroundPosition: '0 0',
      backgroundSize: 'auto 100%',
      height: '39px',
      animation: '250s linear infinite $move-decor'
    },
    balloonOne: {
      position: 'absolute',
      height: '112px',
      width: '89px',
      top: '70px',
      left: '130px',
      animation: '5s ease-in-out infinite $float-balloon-1',
      zIndex: 5,
      backgroundImage: 'url(' + _balloon.default + ')'
    },
    balloonTwo: {
      position: 'absolute',
      height: '70px',
      width: '55px',
      top: '40px',
      left: '60px',
      animation: '6s ease-in-out .2s infinite $float-balloon-2',
      zIndex: 2,
      backgroundImage: 'url(' + _balloon2.default + ')'
    },
    hill: {
      position: 'absolute',
      bottom: 0,
      height: '640px',
      backgroundRepeat: 'repeat-x',
      width: '100%',
      backgroundSize: 'cover',
      backgroundImage: 'url(' + _hill.default + ')'
    },
    town: {
      position: 'absolute',
      bottom: '30px',
      height: '240px',
      display: 'block',
      width: '100000%',
      backgroundRepeat: 'repeat',
      backgroundPosition: '0 0',
      backgroundSize: 'auto 100%',
      animation: '700s linear infinite $move-decor',
      backgroundImage: 'url(' + _town.default + ')'
    },
    cloudOne: {
      position: 'absolute',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      opacity: 0.9,
      zIndex: '3',
      top: '120px',
      height: '98px',
      width: '224px',
      left: '840px',
      backgroundImage: 'url(' + _cloud.default + ')'
    },
    cloudTwo: {
      position: 'absolute',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      opacity: 0.4,
      zIndex: '4',
      top: '110px',
      height: '89px',
      width: '167px',
      left: '750px',
      backgroundImage: 'url(' + _cloud2.default + ')'
    },
    cloudThree: {
      position: 'absolute',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      opacity: 0.4,
      zIndex: '3',
      top: '40px',
      height: '89px',
      width: '167px',
      left: '-167px',
      animation: '43s ease-out infinite $cloud-3',
      backgroundImage: 'url(' + _cloud3.default + ')'
    },
    cloudFour: {
      position: 'absolute',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      opacity: 0.9,
      zIndex: '4',
      top: '60px',
      height: '98px',
      width: '224px',
      left: '-224px',
      animation: '300s ease-out infinite $cloud-4',
      backgroundImage: 'url(' + _cloud4.default + ')'
    }
  };
});
var _default = useStyles;
exports.default = _default;
