// harelkit/lib/components/Select/Option.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Option = _interopRequireDefault(require("./Option.style"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Option = function Option(props) {
  var id = props.id,
      item = props.item,
      selected = props.selected,
      onClick = props.onClick,
      multi = props.multi;
  var isSelected = selected(item);

  var handleClick = function handleClick(e) {
    onClick(item);
  };

  return /*#__PURE__*/_react.default.createElement(_Option.default.Root, {
    id: id,
    role: "option",
    multi: multi,
    selected: isSelected,
    "aria-selected": isSelected,
    "aria-disabled": item.disabled,
    "aria-label": item.label,
    className: item.disabled && "disabled",
    tabIndex: "-1",
    onClick: handleClick
  }, item.label);
};

var _default = Option;
exports.default = _default;
