// common-ui/lib/components/CreditCardField/CreditCardField.constants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DEFAULT_CVC_LENGTH = exports.DEFAULT_CARD_FORMAT = exports.CARD_TYPES = void 0;

var _CreditCardField = _interopRequireDefault(require("./CreditCardField.images"));

var DEFAULT_CVC_LENGTH = 3;
exports.DEFAULT_CVC_LENGTH = DEFAULT_CVC_LENGTH;
var DEFAULT_CARD_FORMAT = /(\d{1,4})/g;
exports.DEFAULT_CARD_FORMAT = DEFAULT_CARD_FORMAT;
var CARD_TYPES = [{
  type: 'amex',
  format: /(\d{1,4})(\d{1,6})?(\d{1,5})?/,
  startPattern: /^3[47]/,
  maxCardNumberLength: 15,
  cvcLength: 4,
  image: _CreditCardField.default.amex
}, {
  type: 'dinersclub',
  format: DEFAULT_CARD_FORMAT,
  startPattern: /^(36|38|30[0-5])/,
  maxCardNumberLength: 14,
  cvcLength: DEFAULT_CVC_LENGTH,
  image: _CreditCardField.default.placeholder
}, {
  type: 'discover',
  format: DEFAULT_CARD_FORMAT,
  startPattern: /^(6011|65|64[4-9]|622)/,
  maxCardNumberLength: 16,
  cvcLength: DEFAULT_CVC_LENGTH,
  image: _CreditCardField.default.placeholder
}, {
  type: 'mastercard',
  format: DEFAULT_CARD_FORMAT,
  startPattern: /^(5[1-5]|677189)|^(222[1-9]|2[3-6]\d{2}|27[0-1]\d|2720)/,
  maxCardNumberLength: 16,
  cvcLength: DEFAULT_CVC_LENGTH,
  image: _CreditCardField.default.mastercard
}, {
  type: 'unionpay',
  format: DEFAULT_CARD_FORMAT,
  startPattern: /^62/,
  maxCardNumberLength: 19,
  cvcLength: DEFAULT_CVC_LENGTH,
  image: _CreditCardField.default.placeholder
}, {
  type: 'visa',
  format: DEFAULT_CARD_FORMAT,
  startPattern: /^4/,
  maxCardNumberLength: 19,
  cvcLength: DEFAULT_CVC_LENGTH,
  image: _CreditCardField.default.visa
}];
exports.CARD_TYPES = CARD_TYPES;
