// common-ui/lib/components/AmountPicker/AmountPicker.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: 'flex',
      paddingTop: 0,
      '& .MuiFormControl-root[class*="MuiTextField-root"] > label + [class*="MuiInput-formControl"]': {
        paddingTop: 0
      }
    },
    stepButtonWrapper: {
      position: 'relative',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0 8px'
    },
    stepButton: {
      border: 'none',
      minWidth: '32px'
    },
    inputText: {
      fontSize: '30px',
      fontWeight: '500',
      fontStretch: 'normal',
      fontStyle: 'normal',
      lineHeight: '0.77',
      letterSpacing: 'normal',
      textAlign: 'center',
      padding: '7px 10px 7.5px 10px',
      direction: 'ltr',
      caretColor: 'transparent',
      '&::-ms-clear': {
        display: 'none'
      }
    },
    inputLabel: {
      fontSize: '0',
      padding: '0'
    },
    accessibilityLabel: {
      fontSize: '0',
      whiteSpace: 'pre'
    }
  };
});
var _default = useStyles;
exports.default = _default;
