// common-ui/lib/components/StripBox/StripBox.style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    stripBox: (0, _defineProperty2.default)({
      overflow: 'hidden',
      background: '#F4F6FC',
      position: 'relative',
      '& + &': {
        marginTop: theme.spacing(3)
      }
    }, theme.breakpoints.up('lg'), {
      display: 'flex'
    }),
    label: {
      display: 'block',
      position: 'absolute',
      right: 0,
      top: '20px',
      background: function background(props) {
        return props.categoryColor ? theme.palette[props.categoryColor].main : theme.palette.primary.main;
      },
      zIndex: 1,
      borderTopLeftRadius: theme.spacing(2),
      borderBottomLeftRadius: theme.spacing(2),
      color: '#fff',
      paddingRight: theme.spacing(2),
      paddingLeft: theme.spacing(1),
      fontSize: '0.9rem'
    },
    header: (0, _defineProperty2.default)({
      flex: '0 0 300px',
      background: function background(props) {
        return props.transparentHeader ? '' : '#fff';
      },
      padding: function padding(props) {
        return props.category ? "40px ".concat(theme.spacing(2), "px ").concat(theme.spacing(2), "px ").concat(theme.spacing(2), "px") : theme.spacing(2);
      },
      margin: theme.spacing(1),
      borderRadius: theme.spacing(0.5),
      '& h3 + h5': {
        paddingTop: theme.spacing(2)
      }
    }, theme.breakpoints.up('xl'), {
      flex: '0 0 340px'
    }),
    content: {
      flex: 1,
      padding: theme.spacing(1)
    }
  };
});
var _default = useStyles;
exports.default = _default;
