// common-ui/lib/components/StripBox/StripBox.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _StripBox = _interopRequireDefault(require("./StripBox.style"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _ = require("../../");
var _clsx = _interopRequireDefault(require("clsx"));
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function StripBox(props) {
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var category = props.category,
    header = props.header,
    subheader = props.subheader,
    children = props.children,
    additionalHeaders = props.additionalHeaders,
    className = props.className;
  var classes = (0, _StripBox.default)(props);
  var hrlBo = props['data-hrl-bo'];
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)('MuiStripBox', classes.stripBox, className),
    "data-hrl-bo": hrlBo
  }, category && /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)('MuiStripBox-label', classes.label),
    "data-hrl-bo": hrlBo + '_category'
  }, category), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.header
  }, header && /*#__PURE__*/_react.default.createElement(_.Typography, {
    variant: "h3",
    "data-hrl-bo": hrlBo + '_header'
  }, header), subheader && /*#__PURE__*/_react.default.createElement(_.Typography, {
    variant: "h5"
  }, subheader), additionalHeaders), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, children));
}
StripBox.propTypes = {
  /** Category side label text */
  category: _propTypes.default.string,
  /** Category side label text color */
  categoryColor: _propTypes.default.oneOf(['primary', 'secondary', 'info', 'warn', 'success', 'error']),
  /** Header text */
  header: _propTypes.default.string,
  /** Is Header background transparent */
  transparentHeader: _propTypes.default.bool,
  /** SubHeader text */
  subheader: _propTypes.default.string,
  /** Children to be drawn inside the main container */
  children: _propTypes.default.any,
  /** Additional Header components */
  additionalHeaders: _propTypes.default.any,
  /**
   * Additional className
   */
  className: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.string]),
  /**
   * Override data-hrl-bo
   */
  'data-hrl-bo': _propTypes.default.string
};
var _default = StripBox;
exports.default = _default;
