// common-ui/lib/components/FileUpload/fileItem/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _colors = require("@material-ui/core/colors");

var styles = {
  container: {
    display: 'flex',
    marginBottom: '12px'
  },
  iconWrapper: {
    marginLeft: '16px',
    height: '40px',
    width: '40px',
    minWidth: '40px',
    color: '#BABABA'
  },
  uploadInfo: {
    width: 'calc( 100% - 56px )',
    flex: '1'
  },
  uploadMeta: {
    display: 'flex',
    alignItems: 'center'
  },
  uploadTitle: {
    display: 'flex',
    flex: '1',
    minWidth: '0',
    overflow: 'hidden',
    outline: 'none'
  },
  uploadFileName: {
    marginLeft: '16px',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    direction: 'ltr',
    lineHeight: '1',
    display: 'block'
  },
  uploadSize: {
    color: _colors.grey[800],
    direction: 'ltr',
    whiteSpace: 'nowrap',
    lineHeight: '1',
    marginLeft: '16px',
    display: 'block'
  },
  uploadActions: {
    marginRight: 'auto',
    textAlign: 'left'
  },
  deleteButton: {
    marginLeft: '-15px'
  },
  error: {
    color: '#c80b0b',
    fontSize: '0.9em',
    margin: '4px 0',
    display: 'block'
  },
  fileName: {
    marginRight: '10px',
    color: _colors.grey[800]
  }
};
var _default = styles;
exports.default = _default;
