// common-ui/lib/components/Tabs/TabPanel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _excluded = ["children", "value", "index"];

function TabPanel(props) {
  var children = props.children,
      value = props.value,
      index = props.index,
      other = (0, _objectWithoutProperties2.default)(props, _excluded);
  return /*#__PURE__*/_react.default.createElement(_Typography.default, (0, _extends2.default)({
    component: "div",
    role: "tabpanel",
    hidden: value !== index,
    id: "tabpanel-".concat(index)
  }, other), /*#__PURE__*/_react.default.createElement(_Box.default, {
    p: 3
  }, children));
}

TabPanel.propTypes = {
  children: _propTypes.default.node,
  index: _propTypes.default.any.isRequired,
  value: _propTypes.default.any.isRequired
};
var _default = TabPanel;
exports.default = _default;
