// common-ui/lib/components/Stepper/Step.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _Step = _interopRequireDefault(require("./Step.style"));

var _clsx8 = _interopRequireDefault(require("clsx"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function Icon(props) {
  var icon = props.icon; // TODO: replace null with default icon ?!

  return icon ? icon : null;
}

function Step(props) {
  var title = props.title,
      icon = props.icon,
      active = props.active,
      done = props.done,
      route = props.route,
      onClick = props.onClick,
      disabled = props.disabled,
      isClickable = props.isClickable,
      isClickableOnlyDoneStep = props.isClickableOnlyDoneStep;
  var hrlBo = props['data-hrl-bo'] || 'atm-step';
  var classes = (0, _Step.default)(props);

  function handleClick(event) {
    if (!!onClick) onClick(route);
  }

  return /*#__PURE__*/_react.default.createElement("li", {
    disabled: disabled,
    "aria-hidden": "true",
    className: (0, _clsx8.default)(classes.root, (0, _defineProperty2.default)({}, classes.done, done), (0, _defineProperty2.default)({}, classes.isClickable, isClickable && !isClickableOnlyDoneStep), (0, _defineProperty2.default)({}, classes.isClickable, done && isClickableOnlyDoneStep)),
    "data-hrl-bo": hrlBo
  }, /*#__PURE__*/_react.default.createElement("div", {
    onClick: handleClick,
    className: (0, _clsx8.default)(classes.icon, (0, _defineProperty2.default)({}, classes.iconActive, active), (0, _defineProperty2.default)({}, classes.iconDone, done))
  }, /*#__PURE__*/_react.default.createElement(Icon, {
    icon: icon
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx8.default)(classes.title, (0, _defineProperty2.default)({}, classes.titleActive, active), (0, _defineProperty2.default)({}, classes.titleDone, done))
  }, title));
}

Step.propTypes = {
  /**
   * Step title
   */
  title: _propTypes.default.string,

  /**
   * Icon representation of step
   */
  icon: _propTypes.default.element.isRequired,

  /**
   * On click function
   */
  onClick: _propTypes.default.func,

  /**
   * Step is clickable
   */
  isClickable: _propTypes.default.bool,

  /**
   * Only Done Step is clickable
   */
  isClickableOnlyDoneStep: _propTypes.default.bool,
  active: _propTypes.default.bool,
  done: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  route: _propTypes.default.any,

  /**
   * Override data-hrl-bo
   */
  'data-hrl-bo': _propTypes.default.string
};
var _default = Step;
exports.default = _default;
