// harelkit/lib/components/Label/Label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Label = _interopRequireDefault(require("./Label.style"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Label = function Label(props) {
  var children = props.children,
      htmlFor = props.htmlFor,
      required = props.required,
      bo = props.bo,
      component = props.component;
  return /*#__PURE__*/_react.default.createElement(_Label.default.Root, {
    htmlFor: htmlFor,
    "data-hrl-bo": bo,
    as: component
  }, children, required && /*#__PURE__*/_react.default.createElement(_Label.default.Asterix, {
    "aria-hidden": "true"
  }, "*"));
};

Label.propTypes = {
  /**
   * Help connect the label to the input for accessibility purpose
   */
  htmlFor: _propTypes.default.string,

  /**
   * If true, add asterix to label
   */
  required: _propTypes.default.bool,

  /**
   * Will add data-hrl-bo attribute to element
   */
  bo: _propTypes.default.string,

  /**
   * Render element type
   */
  component: _propTypes.default.oneOf(['label', 'legend'])
};
Label.defaultProps = {
  required: false,
  component: 'label'
};
var _default = Label;
exports.default = _default;
