// common-ui/lib/components/Checkbox/Checkbox.style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      alignItems: 'flex-start',
      '& [class*="MuiFormControlLabel-label"]': {
        padding: '9px 0 !important',
        lineHeight: '0',
        '& [class*="MuiFormLabel-root"]': {
          letterSpacing: 'normal',
          fontSize: '16px',
          fontStretch: 'normal',
          fontStyle: 'normal',
          lineHeight: '1.25'
        }
      },
      '& [class*="Mui-disabled"] [class*="MuiFormLabel-root"]': {
        color: '#5c6266'
      }
    },
    standaloneRoot: {
      '& [class*="MuiFormLabel-root"]': {
        letterSpacing: 'normal',
        fontSize: '16px',
        fontStretch: 'normal',
        fontStyle: 'normal',
        lineHeight: '1.25'
      }
    }
  };
});
var _default = useStyles;
exports.default = _default;
