// harelkit/lib/components/Tooltip/Tooltip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _misc = require("../../utils/misc");

var _core = require("@material-ui/core");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var noop = function noop() {};

var Tooltip = function Tooltip(props) {
  var pickedProps = (0, _misc.pick)(['children', 'title', 'open', 'onOpen', 'onClose', 'placement', 'bo'], props);
  pickedProps.TransitionProps = {
    'data-hrl-bo': pickedProps.bo ? "".concat(pickedProps.bo, "-tooltip") : null
  };
  return /*#__PURE__*/_react.default.createElement(_core.Tooltip, _extends({}, pickedProps, {
    arrow: true
  }));
};

Tooltip.propTypes = {
  /**
   * Tooltip reference element. Needs to be able to hold a ref
   * */
  children: _propTypes.default.node,

  /**
   *  Tooltip title. Zero-length titles string are never displayed
   * */
  title: _propTypes.default.node,

  /**
   * If true, the tooltip is shown
   * */
  open: _propTypes.default.bool,

  /**
   * Callback fired when the component requests to be open.
   * function(event: object) => void
   */
  onOpen: _propTypes.default.func,

  /**
   * Callback fired when the component requests to be closed.
   * function(event: object) => void
   */
  onClose: _propTypes.default.func,

  /**
   * Tooltip placement
   * */
  placement: _propTypes.default.oneOf(['bottom', 'right', 'top', 'left']),

  /**
   * Will add data-hrl-bo attribute to element.
   * 
   * bo attribute:
   * {bo}-tooltip
   */
  bo: _propTypes.default.string
};
Tooltip.defaultProps = {
  placement: 'bottom',
  onOpen: noop,
  onClose: noop
};
var _default = Tooltip;
exports.default = _default;
