// common-ui/lib/components/BaseLayoutProvider/Themes/base/base.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var spacing = 8;
var direction = 'rtl';
var shape = {};
/**
 * Palette
 */

var WHITE = '#FFFFFF';
var BLACK = '#000000';
var palette = {
  primary: {
    light: '#1A599A',
    main: '#003f80',
    dark: '#002667',
    contrastText: WHITE,
    text: WHITE,
    lightText: WHITE
  },
  secondary: {
    superLight: '#e4e9f2',
    light: '#339FE5',
    main: '#006cb2',
    dark: '#005399',
    text: WHITE,
    contrastText: WHITE,
    lightText: BLACK
  },
  error: {
    superLight: '#ffe5db',
    light: '#F54437',
    main: '#DB2A1D',
    dark: '#C21104',
    contrastText: WHITE,
    text: WHITE,
    lightText: BLACK
  },
  warn: {
    light: '#FFD442',
    main: '#FBBA28',
    dark: '#E2A10F',
    contrastText: BLACK,
    text: BLACK,
    lightText: BLACK
  },
  info: {
    light: '#36A3E2',
    main: '#0370af',
    dark: '#005796',
    contrastText: WHITE,
    text: WHITE,
    lightText: BLACK
  },
  success: {
    light: '#67BC57',
    main: '#4DA23D',
    dark: '#348924',
    contrastText: BLACK,
    text: BLACK,
    lightText: BLACK
  },
  background: {
    default: WHITE
  },
  text: {
    primary: '#003f80',
    secondary: '#006cb2',
    disabled: '#1D8AC9',
    hint: 'rgba(0, 63, 128, 0.38)'
  }
};
/**
 * Typography
 */

var typography = {
  fontFamily: '"mfw_protocolharel", Arial',
  fontSize: 15,
  htmlFontSize: 15
};
/**
 * Breakpoints
 */

var breakpoints = {
  keys: ['xs', 'sm', 'md', 'lg', 'xl'],
  values: {
    xs: 0,
    sm: 518,
    md: 765,
    lg: 1013,
    xl: 1440
  }
};
/**
 * Overrides
 */

var overrides = {
  MuiIconButton: {
    root: {
      padding: "".concat(spacing, "px"),
      '&.Mui-focusVisible': {
        backgroundColor: 'rgba(0, 108, 178, 0.32)'
      }
    }
  },
  MuiContainer: {
    maxWidthXl: {
      '@media(minWidth: 1400px)': {
        width: '1400px'
      }
    }
  },
  MuiToggle: {
    margin: 0
  },
  MuiSnackbarContent: {
    root: {
      backgroundColor: '#fff',
      color: palette.primary.main
    },
    action: {
      marginRight: 'auto',
      marginLeft: "-".concat(spacing, "px"),
      paddingRight: "".concat(spacing * 2, "px"),
      paddingLeft: null
    }
  },
  MuiBadge: {
    badge: {
      color: '#ffffff !important'
    }
  },
  MuiDialog: {
    paper: {
      overflow: 'hidden'
    }
  },
  MuiDialogActions: {
    root: {
      padding: "".concat(spacing * 2, "px ").concat(spacing * 3, "px")
    }
  },
  MuiButtonBase: {
    root: {
      border: '1px solid transparent'
    }
  },
  MuiRangeDatepicker: {
    dayTextColor: 'rgba(0, 63, 128, 1)',
    highlightDayTextColor: palette.primary.main,
    hoveredDayTextColor: palette.primary.main,
    trailColor: '#e7f4e0',
    trailHoverColor: '#e7f4e0',
    fromToColor: '#cfeac2'
  },
  MuiListItem: {
    root: {
      textAlign: 'right'
    }
  },
  MuiLink: {
    underlineHover: {
      '&:hover': {
        textDecorationSkipInk: 'none'
      }
    },
    underlineAlways: {
      textDecorationSkipInk: 'none'
    }
  },
  MuiButton: {
    label: {
      lineHeight: 1
    },
    root: {
      '&:not(:last-child)': {
        marginLeft: '16px'
      },
      fontSize: '0.9rem',
      '&:focus': {
        boxShadow: '0 0 5 rgba(0,0,0,0.3) !important',
        border: '1px solid rgba(255,255,255,0.5)'
      },
      padding: '6px 16px',
      borderRadius: '4px',
      color: '#003f80',
      textTransform: 'initial'
    },
    contained: {
      boxShadow: 'none',
      '&:active': {
        boxShadow: 'none'
      },
      '&$focusVisible': {
        boxShadow: 'none'
      }
    },
    outlined: {
      background: '#fff',
      padding: '6px 16px',
      '&:focus': {
        border: '1px solid rgba(255,255,255,0.5)'
      }
    },
    outlinedPrimary: {
      '&:focus': {
        border: '1px solid rgba(0, 63, 128, 0.5)'
      }
    },
    outlinedSecondary: {
      '&:focus': {
        border: '1px solid rgba(0, 108, 178, 0.5)'
      }
    },
    containedPrimary: {
      backgroundColor: '#002667',
      '&:hover': {
        backgroundColor: '#002667'
      }
    }
  },
  MuiInputLabel: {
    root: {
      transformOrigin: 'top right',
      pointerEvents: 'none',
      fontWeight: 500
    },
    formControl: {
      left: 'auto',
      right: spacing * 0.5,
      transform: 'translate(0, 19px)',
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      width: '100%'
    },
    shrink: {
      transformOrigin: 'top right',
      width: '125%'
    },
    '&.Mui-focused': {
      fontWeight: 100
    }
  },
  MuiTooltip: {
    arrow: {
      '&[x-placement*="left"] $arrow': {
        /* tricky doubly-quoted empty string so mui parses it as truly empty */
        content: '""',
        display: 'block',
        width: '0',
        height: '0',
        color: 'red',
        position: 'absolute',
        borderLeft: '10px solid transparent',
        borderRight: '10px solid transparent',

        /* border color should probably match whatever your tooltip color is */
        borderTop: '10px solid #000',
        left: 'calc(50% - 10px)',
        top: '100%'
      }
    }
  },
  MuiTextField: {
    root: {
      width: '100%'
    }
  },
  MuiFormHelperText: {
    root: {
      textAlign: 'right',
      marginTop: 2,
      color: '#5c6266'
    }
  },
  MuiInputBase: {
    root: {
      '&.Mui-disabled': {
        color: '#777777'
      },
      '& > input': {
        borderColor: 'transparent'
      }
    },
    inputAdornedStart: {
      marginRight: '8px'
    }
  },
  MuiChip: {
    deleteIcon: {
      margin: '0 -5px 0 6px'
    },
    root: {
      minHeight: '32px',
      height: 'auto '
    },
    label: {
      whiteSpace: 'normal'
    }
  },
  MuiMenuItem: {
    root: {
      fontSize: '1rem',
      minHeight: null,
      lineHeight: 1.5
    }
  },
  MuiList: {
    root: {
      direction: 'rtl'
    },
    padding: {
      paddingTop: '4px',
      paddingBottom: '4px'
    }
  },
  MuiTabs: {
    root: {
      marginTop: spacing * 3,
      marginBottom: spacing * 3
    },
    scroller: {
      '&:after': {
        content: '""',
        display: 'block',
        width: '100%',
        borderBottom: '3px solid #E3EBF6'
      }
    },
    indicator: {
      borderBottom: '3px solid #0088C9'
    },
    vertical: {
      borderLeft: '1px solid #E3EBF6',
      '& [class*="MuiButtonBase-root"]': {
        padding: '10px 16px'
      },
      '& [class*="MuiTabs-indicator"]': {
        right: 'auto',
        left: 0
      },
      '& [class*="MuiTabs-scroller"]': {
        '&:after': {
          content: '""',
          display: 'none',
          width: '100%',
          borderBottom: '0px solid #E3EBF6'
        }
      },
      '& [class*="MuiTabs-scrollButtons"]': {
        '& svg': {
          transform: 'rotate(-90deg)'
        }
      }
    }
  },
  MuiTab: {
    root: {
      minHeight: null,
      padding: '20px 16px',
      fontSize: '1.1rem',
      '&:focus': {
        backgroundColor: 'rgba(0, 63, 128, 0.08)'
      }
    }
  },
  MuiSelect: {
    select: {
      paddingLeft: '32px',
      paddingRight: '4px !important',
      backgroundColor: 'none',
      borderRadius: "".concat(spacing, "px ").concat(spacing, "px 0 0"),
      '&:focus': {
        borderRadius: spacing,
        backgroundColor: 'none'
      },
      '&[class*="MuiSelect-selectMenu"]': {
        paddingRight: '4px !important'
      }
    },
    icon: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    },
    iconFilled: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    }
  },
  MuiNativeSelect: {
    root: {
      width: '100%'
    },
    select: {
      paddingLeft: '32px',
      paddingRight: '4px !important',
      backgroundColor: 'none',
      borderRadius: "".concat(spacing, "px ").concat(spacing, "px 0 0"),
      '&:focus': {
        borderRadius: spacing,
        backgroundColor: 'none'
      },
      '&[class*="MuiSelect-selectMenu"]': {
        paddingRight: '4px !important'
      }
    },
    icon: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    },
    iconFilled: {
      left: 6,
      right: 'auto',
      top: 'calc(50% - 6px)'
    }
  },
  MuiFormLabel: {
    root: {
      color: '#003f80',
      fontSize: '1.2rem',
      zIndex: 2,
      padding: 0,
      paddingLeft: 4,
      fontWeight: '500',
      '&.Mui-disabled': {
        color: '#5c6266'
      }
    },
    asterisk: {
      color: '#D32704',
      letterSpacing: -2
    }
  },
  MuiCheckbox: {
    root: {
      color: '#DDE4ED'
    },
    colorSecondary: {
      color: '#2D76CB',
      '&.Mui-disabled': {
        color: '#5c6266'
      }
    }
  },
  MuiExpansionPanelSummary: {
    root: {
      padding: '0 16px 0 0',
      background: '#E3EBF6',
      '&$expanded': {
        minHeight: '48px'
      },
      content: {
        '&$expanded': {
          margin: '12px 0'
        }
      }
    }
  },
  MuiSvgIcon: {
    colorAction: {
      color: '#0088C9'
    }
  },
  MuiExpansionPanelDetails: {
    root: {
      padding: '16px'
    }
  },
  MuiSwitch: {
    input: {
      left: null,
      width: null
    }
  },
  MuiFormControl: {
    root: {
      width: '100%'
    }
  },
  MuiFormControlLabel: {
    root: {
      marginRight: 'auto',
      marginLeft: 'auto'
    },
    labelPlacementStart: {
      marginLeft: null,
      marginRight: null
    },
    label: {
      fontSize: '1.2rem',
      fontWeight: 500,
      '&.Mui-disabled': {
        color: '#5c6266'
      }
    }
  },
  MuiInputAdornment: {
    positionStart: {
      marginRight: '3.5px'
    }
  },
  MuiDivider: {
    root: {
      backgroundColor: '#E3EBF6',
      height: '3px',
      borderRadius: '4px'
    }
  },
  MuiTypography: {
    h1: {
      fontSize: '1.7rem'
    },
    h2: {
      fontSize: '1.4rem'
    },
    h3: {
      fontSize: '1.3rem'
    },
    h4: {
      fontSize: '1.2rem'
    },
    h5: {
      fontSize: '1.1rem'
    },
    h6: {
      fontSize: '0.9rem'
    },
    body1: {
      fontSize: '1rem'
    },
    body2: {
      fontSize: '0.9rem'
    }
  },
  MuiInput: {
    input: {
      fontWeight: '500',
      padding: '3.5px'
    }
  },
  MuiAutocomplete: {
    hasClearIcon: {
      '& .MuiInputBase-root.MuiInput-formControl': {
        paddingRight: 0
      },
      '&[class*="MuiAutocomplete-hasPopupIcon"] [class*="MuiAutocomplete-inputRoot"]': {
        paddingRight: '0 !important'
      }
    },
    clearIndicator: {
      padding: '1px'
    },
    popupIndicator: {
      padding: '0px',
      border: '0px'
    },
    inputRoot: {
      paddingRight: 0,
      paddingLeft: 32,
      '&[class*="MuiInput-root"]': {
        paddingRight: 0
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]': {
        borderColor: 'transparent'
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child': {
        padding: '4px !important'
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child::-ms-clear': {
        display: 'none',
        width: 0,
        height: 0
      },
      '&[class*="MuiInput-root"] [class*="MuiAutocomplete-input"]:first-child::-ms-reveal': {
        display: 'none',
        width: 0,
        height: 0
      },
      '&[class*="MuiFilledInput-adornedEnd"] [class*="MuiAutocomplete-endAdornment"]': {
        left: 7,
        right: 'auto !important'
      }
    },
    endAdornment: {
      left: 0,
      right: 'auto'
    }
  },
  MuiPickersYear: {
    root: {
      '&:focus': {
        border: '1px solid ' + _core.colors.grey[400]
      }
    }
  },
  MuiPickersMonth: {
    root: {
      '&:focus': {
        border: '1px solid ' + _core.colors.grey[400]
      }
    }
  },
  MuiCircularProgress: {
    root: {
      '-ms-overflow-x': 'hidden',
      '-ms-overflow-y': 'hidden'
    }
  },
  MuiGrid: {
    item: {
      position: 'relative'
    }
  },
  MuiPickersCalendarHeader: {
    dayLabel: {
      color: 'rgba(0, 0, 0, 0.54)',
      fontWeight: 'bold'
    }
  },
  MuiPickersToolbarText: {
    toolbarTxt: {
      color: '#FFFFFF'
    }
  }
};
var _default = {
  palette: palette,
  spacing: spacing,
  typography: typography,
  direction: direction,
  breakpoints: breakpoints,
  overrides: overrides,
  shape: shape
};
exports.default = _default;
