// common-ui/lib/components/FileUpload/constants/constants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WHITE_LIST_REGEX = exports.VALID_MESSAGE = exports.TRASH_CAN_INTERIOR_IMAGE_PATH = exports.TRASH_CAN_EXTERIOR_IMAGE_PATH = exports.SELECT_PLACE_HOLDER = exports.REMOVED_TAG = exports.ONLY_ONE_FILE_FOR_UNIQUE_TAG_MESSAGE = exports.MANDATORY_FILES_ARE_MISSING_MESSAGE = exports.ICONS_MAP = exports.EXCEL_EXTENSIONS = exports.DROP_ZONE_PLUS_IMAGE_PATH = exports.DRAG_FILES_HERE_TEXT = exports.DEFAULT_MAX_FILES = exports.DEFAULT_EXTENSIONS = exports.CLEARED_ALL_TAGS = exports.CHOSE_FILES_TO_UPLOAD_TEXT = exports.CHOOSE_ATLEAST_ONE_FILE_TYPE_MESSAGE = exports.ADDED_A_TAG = void 0;

var _PictureAsPdfTwoTone = _interopRequireDefault(require("@material-ui/icons/PictureAsPdfTwoTone"));

var _ImageTwoTone = _interopRequireDefault(require("@material-ui/icons/ImageTwoTone"));

var _DescriptionTwoTone = _interopRequireDefault(require("@material-ui/icons/DescriptionTwoTone"));

var ICONS_MAP = {
  docx: _DescriptionTwoTone.default,
  doc: _DescriptionTwoTone.default,
  pdf: _PictureAsPdfTwoTone.default,
  xls: _DescriptionTwoTone.default,
  xlsx: _DescriptionTwoTone.default,
  tiff: _ImageTwoTone.default,
  bmp: _ImageTwoTone.default,
  gif: _ImageTwoTone.default,
  jpeg: _ImageTwoTone.default,
  png: _ImageTwoTone.default,
  jpg: _ImageTwoTone.default,
  default: _ImageTwoTone.default
};
exports.ICONS_MAP = ICONS_MAP;
var DEFAULT_EXTENSIONS = ['docx', 'doc', 'pdf', 'xls', 'xlsx', 'tiff', 'bmp', 'gif', 'jpeg', 'png', 'jpg'];
exports.DEFAULT_EXTENSIONS = DEFAULT_EXTENSIONS;
var WHITE_LIST_REGEX = /[^!'(){}./? \\,:@~\r\n"\[\]_a-zA-Z0-9\u0590-\u05FF-]/g;
exports.WHITE_LIST_REGEX = WHITE_LIST_REGEX;
var DEFAULT_MAX_FILES = 5;
exports.DEFAULT_MAX_FILES = DEFAULT_MAX_FILES;
var EXCEL_EXTENSIONS = ['xls', 'xlt', 'xlm', 'xlsx', 'xlsm', 'xltx', 'xltm', 'xlsb', 'xla', 'xlam', 'xll', 'xll'];
exports.EXCEL_EXTENSIONS = EXCEL_EXTENSIONS;
var REMOVED_TAG = 'remove-value';
exports.REMOVED_TAG = REMOVED_TAG;
var ADDED_A_TAG = 'select-option';
exports.ADDED_A_TAG = ADDED_A_TAG;
var CLEARED_ALL_TAGS = 'clear';
exports.CLEARED_ALL_TAGS = CLEARED_ALL_TAGS;
var DROP_ZONE_PLUS_IMAGE_PATH = 'M16 2C8.27 2 2 8.27 2 16s6.27 14 14 14 14-6.27 14-14S23.73 2 16 2zm0 26C9.38 28 4 22.62 4 16S9.38 4 16 4s12 5.38 12 12-5.38 12-12 12zm6-13v2h-5v5h-2v-5h-5v-2h5v-5h2v5h5z';
exports.DROP_ZONE_PLUS_IMAGE_PATH = DROP_ZONE_PLUS_IMAGE_PATH;
var TRASH_CAN_EXTERIOR_IMAGE_PATH = 'M27.14 34H8.86A2.93 2.93 0 0 1 6 31V11.23h2V31a.93.93 0 0 0 .86 1h18.28a.93.93 0 0 0 .86-1V11.23h2V31a2.93 2.93 0 0 1-2.86 3zM30.78 9H5a1 1 0 0 1 0-2h25.78a1 1 0 0 1 0 2z';
exports.TRASH_CAN_EXTERIOR_IMAGE_PATH = TRASH_CAN_EXTERIOR_IMAGE_PATH;
var TRASH_CAN_INTERIOR_IMAGE_PATH = 'M21 13h2v15h-2zM13 13h2v15h-2zM23 5.86h-1.9V4h-6.2v1.86H13V4a2 2 0 0 1 1.9-2h6.2A2 2 0 0 1 23 4z';
exports.TRASH_CAN_INTERIOR_IMAGE_PATH = TRASH_CAN_INTERIOR_IMAGE_PATH;
var CHOSE_FILES_TO_UPLOAD_TEXT = ' בחר קבצים להעלאה';
exports.CHOSE_FILES_TO_UPLOAD_TEXT = CHOSE_FILES_TO_UPLOAD_TEXT;
var DRAG_FILES_HERE_TEXT = 'ניתן לגרור קבצים לכאן או לבחור קבצים להעלאה';
exports.DRAG_FILES_HERE_TEXT = DRAG_FILES_HERE_TEXT;
var SELECT_PLACE_HOLDER = 'לחצ/י לבחירת אפשרויות';
exports.SELECT_PLACE_HOLDER = SELECT_PLACE_HOLDER;
var VALID_MESSAGE = 'תקין';
exports.VALID_MESSAGE = VALID_MESSAGE;
var MANDATORY_FILES_ARE_MISSING_MESSAGE = 'חסרים מסמכי חובה';
exports.MANDATORY_FILES_ARE_MISSING_MESSAGE = MANDATORY_FILES_ARE_MISSING_MESSAGE;
var CHOOSE_ATLEAST_ONE_FILE_TYPE_MESSAGE = 'יש לבחור סוג מסמך אחד או יותר לכל קובץ שהועלה';
exports.CHOOSE_ATLEAST_ONE_FILE_TYPE_MESSAGE = CHOOSE_ATLEAST_ONE_FILE_TYPE_MESSAGE;
var ONLY_ONE_FILE_FOR_UNIQUE_TAG_MESSAGE = 'עבור סוג מסמך שנבחר ניתן להעלות רק קובץ אחד';
exports.ONLY_ONE_FILE_FOR_UNIQUE_TAG_MESSAGE = ONLY_ONE_FILE_FOR_UNIQUE_TAG_MESSAGE;
