// common-ui/lib/components/Stepper/Step.style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root, _icon, _iconActive;
  return {
    root: (_root = {
      opacity: function opacity(props) {
        return props.disabled ? 0.4 : 1;
      },
      pointerEvents: function pointerEvents(props) {
        return props.disabled ? 'none' : 'auto';
      },
      transition: 'all .1s',
      textAlign: 'center',
      position: 'relative',
      display: 'flex',
      alignItems: 'center',
      padding: "".concat(theme.spacing(0.5), "px ").concat(theme.spacing(1), "px"),
      flexDirection: 'column',
      flex: 1
    }, (0, _defineProperty2.default)(_root, theme.breakpoints.down('sm'), {
      paddingRight: theme.spacing(0.5),
      paddingLeft: theme.spacing(0.5)
    }), (0, _defineProperty2.default)(_root, '&:not(:last-child):after', {
      content: '""',
      display: 'block',
      position: 'absolute',
      width: '100%',
      right: '50%',
      borderTop: '3px solid rgba(223, 230, 239, 0.7)',
      top: '30px'
    }), _root),
    isClickable: {
      cursor: 'pointer',
      '&:hover div:first-child': {
        boxShadow: '0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)'
      }
    },
    done: {
      '&:not(:last-child):after': {
        borderTopColor: '#73CC7D'
      },
      '& div:first-child': {
        transition: '.3s all'
      }
    },
    icon: (_icon = {
      width: theme.spacing(6),
      height: theme.spacing(6),
      borderRadius: '50%',
      backgroundColor: '#fff',
      zIndex: 1,
      border: 'solid 2px #e3ebf6',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      color: theme.palette.primary.main
    }, (0, _defineProperty2.default)(_icon, theme.breakpoints.down('sm'), {
      width: theme.spacing(2),
      height: theme.spacing(2),
      position: 'relative',
      top: '18px'
    }), (0, _defineProperty2.default)(_icon, '& > svg', (0, _defineProperty2.default)({
      height: '36px',
      width: '36px'
    }, theme.breakpoints.down('sm'), {
      display: 'none'
    })), _icon),
    title: (0, _defineProperty2.default)({
      marginTop: theme.spacing(1),
      fontWeight: 300,
      fontSize: '0.9rem',
      padding: theme.spacing(0.5)
    }, theme.breakpoints.down('sm'), {
      display: 'none'
    }),
    titleActive: (0, _defineProperty2.default)({
      backgroundColor: '#FFF5E3',
      fontWeight: 'normal',
      border: '1px solid #F7B208',
      borderRadius: '8px',
      position: 'relative',
      '&:after': {
        content: '""',
        width: '8px',
        height: '8px',
        position: 'absolute',
        top: '-4px',
        left: '50%',
        transform: 'translateX(-50%) rotate(45deg)',
        backgroundColor: '#fff5e3',
        borderLeft: '1px solid #F7B208',
        borderTop: '1px solid #F7B208'
      }
    }, theme.breakpoints.down('sm'), {
      display: 'block'
    }),
    iconActive: (_iconActive = {
      width: theme.spacing(6),
      height: theme.spacing(6),
      borderColor: '#F7B208',
      backgroundColor: '#FFF5E3',
      color: '#DE9213'
    }, (0, _defineProperty2.default)(_iconActive, theme.breakpoints.down('sm'), {
      position: 'static'
    }), (0, _defineProperty2.default)(_iconActive, '& > svg', {
      display: 'block'
    }), _iconActive),
    iconDone: {
      borderColor: '#73CC7D',
      backgroundColor: '#EBF6EC',
      color: '#52BC5E'
    }
  };
});
var _default = useStyles;
exports.default = _default;
