// common-ui/lib/components/NextStepper/NextStepper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _NextStepperStep = _interopRequireDefault(require("./NextStepperStep"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: 'relative',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      paddingBottom: '5px'
    }
  };
});

var NextStepper = function NextStepper(props) {
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var classes = useStyles();
  var steps = props.steps,
      onClick = props.onClick,
      classesSteps = props.classesSteps;

  var itemOnClick = function itemOnClick(item) {
    return onClick && onClick(item);
  };

  var currentActiveStepIndex = steps.findIndex(function (s) {
    return s.isActive === true;
  });
  var stepsLength = steps.length;
  var hrlBo = props['data-hrl-bo'] || 'next-stepper-container';
  /**
   * Default is true
   * @param isClickable
   * @returns {boolean}
   */

  var getIsClickable = function getIsClickable(isClickable) {
    if (typeof isClickable === 'boolean') {
      return isClickable;
    }

    return true;
  };

  var setStepsAriaLabel = function setStepsAriaLabel(isActive, index, title, stepsLength) {
    var titleText = '';

    if (title) {
      if (isActive) {
        titleText = " - \u05D4\u05E9\u05DC\u05D1 \u05D4\u05E0\u05D5\u05DB\u05D7\u05D9, ".concat(title);
      } else {
        titleText = " - \u05D4\u05E9\u05DC\u05D1 ".concat(title, " ");
      }
    }

    var stepsTitle = "\u05E9\u05DC\u05D1 ".concat(index + 1, " \u05DE\u05EA\u05D5\u05DA ").concat(stepsLength);
    return stepsTitle + titleText;
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root,
    role: "region",
    "data-hrl-bo": hrlBo,
    "aria-label": "\u05E9\u05DC\u05D1\u05D9 \u05D4\u05EA\u05D4\u05DC\u05D9\u05DA"
  }, steps.map(function (step, i) {
    return /*#__PURE__*/_react.default.createElement(_NextStepperStep.default, (0, _extends2.default)({
      label: setStepsAriaLabel(currentActiveStepIndex === i, i, step.title, stepsLength),
      clickable: !!onClick
    }, getIsClickable(step.isClickable) && {
      onClick: function onClick() {
        return itemOnClick(step);
      }
    }, {
      // onClick={() => getIsClickable(step.isClickable) ? itemOnClick(step) : undefined}
      key: step.title || i,
      title: step.title,
      isDone: step.isDone,
      isActive: step.isActive,
      isLast: i === stepsLength - 1,
      totalSteps: stepsLength,
      currentStep: i,
      icon: step.icon,
      disabled: step.disabled,
      "data-hrl-bo": "next-stepper-step-".concat(i),
      classesSteps: classesSteps,
      isClickable: getIsClickable(step.isClickable)
    }));
  }));
};

NextStepper.propTypes = {
  /** Array containing the stepper items */
  steps: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    isDone: _propTypes.default.bool,
    isActive: _propTypes.default.bool,
    icon: _propTypes.default.node,
    disabled: _propTypes.default.bool,
    isClickable: _propTypes.default.bool
  })),

  /** Stepper item onClick event */
  onClick: _propTypes.default.func,
  classesSteps: _propTypes.default.any,

  /**
   * Data hrl bo attribute for automation testing.
   */
  'data-hrl-bo': _propTypes.default.string
};
var _default = NextStepper;
exports.default = _default;
