// common-ui/lib/components/NextStepper/NextStepperStep.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _commonIcons = require("common-icons");

var _clsx = _interopRequireDefault(require("clsx"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: (0, _defineProperty2.default)({
      position: 'relative',
      backgroundColor: '#fff',
      borderRadius: theme.spacing(10),
      border: '1px solid #003c7f',
      height: 40,
      width: 40,
      boxShadow: '0 4px 9px 0 rgba(119, 119, 119, 0.3)',
      margin: 0,
      cursor: function cursor(props) {
        return !!props.isClickable ? 'pointer' : 'default';
      },
      textAlign: 'center',
      lineHeight: '54px',
      transition: 'all 0.3s',
      '&:hover': {
        backgroundColor: function backgroundColor(_ref) {
          var isClickable = _ref.isClickable;
          return isClickable ? theme.palette.primary.main : 'inherit';
        },
        color: function color(_ref2) {
          var isClickable = _ref2.isClickable;
          return isClickable ? '#fff' : 'inherit';
        }
      },
      '&:hover .divTitleHover': {
        display: function display(_ref3) {
          var isClickable = _ref3.isClickable;
          return isClickable ? 'block' : 'none';
        }
      }
    }, theme.breakpoints.down('sm'), {
      height: 12,
      width: 12,
      boxShadow: 'none',
      color: 'transparent',
      margin: '0 4px',
      border: '1.5px solid #fff',
      '& svg': {
        display: 'none'
      },
      '&:hover': {
        cursor: 'default',
        backgroundColor: 'transparent'
      },
      '&:hover .divTitleHover': {
        display: 'none'
      }
    }),
    nextArrow: (0, _defineProperty2.default)({
      lineHeight: '48px',
      width: 25,
      height: 1,
      backgroundColor: theme.palette.primary.main
    }, theme.breakpoints.down('sm'), {
      display: 'none'
    }),
    done: {
      backgroundColor: '#003c7f',
      color: '#fff',
      pointerEvents: 'none',
      lineHeight: '42px'
    },
    active: (0, _defineProperty2.default)({
      backgroundColor: '#f7b207',
      border: '2px solid #003c7f',
      lineHeight: '52px'
    }, theme.breakpoints.down('sm'), {
      border: '1.5px solid #003c7f',
      height: 10,
      width: 10
    }),
    disabled: (0, _defineProperty2.default)({
      pointerEvents: 'none',
      backgroundColor: '#fff',
      border: '1px solid #848484',
      color: '#848484'
    }, theme.breakpoints.down('sm'), {
      border: 0,
      backgroundColor: '#848484'
    }),
    hoverTitle: {
      display: 'none',
      position: 'absolute',
      color: theme.palette.primary.main,
      top: 50,
      lineHeight: '1',
      width: '100%'
    }
  };
});

var Step = function Step(props) {
  var classes = useStyles(props);
  var label = props.label,
      title = props.title,
      Icon = props.icon,
      isActive = props.isActive,
      isDone = props.isDone,
      isLast = props.isLast,
      isClickable = props.isClickable,
      totalSteps = props.totalSteps,
      currentStep = props.currentStep,
      disabled = props.disabled,
      onClick = props.onClick,
      classesSteps = props.classesSteps;
  var hrlBo = props['data-hrl-bo'] || 'next-stepper-step';
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    role: "img",
    title: title,
    onClick: onClick,
    "aria-label": label,
    "data-hrl-bo": hrlBo,
    "data-hrl-total": totalSteps !== null && totalSteps !== void 0 ? totalSteps : 'unknown',
    "data-hrl-index": currentStep !== null && currentStep !== void 0 ? currentStep : 'unknown',
    "data-hrl-active": (!!isActive).toString(),
    "data-hrl-disabled": (!!disabled).toString(),
    "data-hrl-last": (!!isLast).toString(),
    "data-hrl-done": (!!isDone).toString(),
    className: (0, _clsx.default)(classes.root, classesSteps, isActive && classes.active, isDone && classes.done, disabled && classes.disabled)
  }, isDone && /*#__PURE__*/_react.default.createElement(_commonIcons.IconCheck, null), !isDone && /*#__PURE__*/_react.default.createElement(Icon, {
    fontSize: "default"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.hoverTitle + ' divTitleHover'
  }, title)), !isLast && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.nextArrow
  }));
};

Step.propTypes = {
  label: _propTypes.default.string,
  title: _propTypes.default.string,
  icon: _propTypes.default.node,
  isActive: _propTypes.default.bool,
  isDone: _propTypes.default.bool,
  isLast: _propTypes.default.bool,
  totalSteps: _propTypes.default.number,
  currentStep: _propTypes.default.number,
  disabled: _propTypes.default.bool,
  classesSteps: _propTypes.default.any,

  /** Stepper item onClick event */
  onClick: _propTypes.default.func,

  /**
   * Data hrl bo attribute for automation testing.
   */
  'data-hrl-bo': _propTypes.default.string,

  /**
   * Enable / Disable click
   */
  isClickable: _propTypes.default.bool
};
var _default = Step;
exports.default = _default;
