// harelkit/lib/components/ThemeProvider/Themes/base/palette.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var palette = {
  primary: {
    light5: '#f2f7fc',
    light4: '#e2ecf9',
    light3: '#c7e1ff',
    light2: '#6099dc',
    light: '#4387d5',
    main: '#2d76cb',
    dark: '#2b71c3',
    dark2: '#2769b4',
    dark3: '#2561a7',
    dark4: '#003c7f'
  },
  secondary: {
    light4: '#fef0cd',
    light3: '#fbd983',
    light2: '#f9c239',
    light: '#f9ba20',
    main: '#f7b207',
    dark: '#e4a406',
    dark2: '#d59a06',
    dark3: '#b78405',
    dark4: '#7c5904'
  },
  triadic: {
    light4: '#d7f4db',
    light3: '#61d16e',
    light2: '#33b242',
    light: '#2d9d3a',
    main: '#278732',
    dark: '#268330',
    dark2: '#247f2e',
    dark3: '#206f29',
    dark4: '#1a5b22'
  },
  error: {
    main: '#d32704'
  },
  gray: {
    white: '#fff',
    light2: '#f2f3f4',
    light1: '#d4d6d8',
    primary: '#5c6266',
    dark1: '#383c3e',
    black: '#000'
  },
  text: {
    primary: '#003c7f',
    // 'rgba(0, 0, 0, 0.87)',
    secondary: 'rgba(0, 0, 0, 0.54)',
    disabled: 'rgba(0, 0, 0, 0.38)',
    hint: 'rgba(0, 0, 0, 0.38)',
    divider: 'rgba(0, 0, 0, 0.12)'
  }
};
var _default = palette;
exports.default = _default;
