// common-ui/lib/components/TooltipedComponent/TooltipedComponent.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    componentWrapper: {
      position: 'relative'
    },
    componentGrid: {
      color: theme.palette.primary.main
    },
    xIcon: {
      position: 'absolute',
      left: '-3px',
      top: '-3px',
      fontSize: '24px',
      fontWeight: 'light',
      cursor: 'pointer',
      color: theme.palette.primary.main
    },
    iconWrapper: {
      display: 'inline-block',
      textAlign: 'center',
      cursor: 'pointer'
    },
    icon: {
      boxShadow: '0 2px 6px 0 rgba(119, 119, 119, 0.3), 0 2px 6px 0 rgba(119, 119, 119, 0.3)',
      border: "solid 1px ".concat(theme.palette.primary.main),
      height: '20px',
      width: '20px',
      backgroundColor: '#f3f6f8',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: '50%',
      color: theme.palette.primary.main
    }
  };
});
var _default = useStyles;
exports.default = _default;
