// common-ui/lib/components/Signature/Signature.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _styles = require("@material-ui/core/styles");
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    sigContainer: {
      width: function width(props) {
        return props.width;
      },
      // width: '100%',
      height: '100%',
      margin: '0 auto'
      // backgroundColor: '#fff',
    },

    sigPad: {
      width: function width(props) {
        return props.width ? props.width : '100%';
      },
      height: function height(props) {
        return props.height ? props.height : '100%';
      },
      border: '1px solid lightgray',
      borderRadius: '5px'
    },
    sigButtonContainer: {
      width: function width(props) {
        return props.width;
      },
      textAlign: 'left'
    },
    sigButton: {
      marginTop: '5px',
      '&:focus': {
        border: 'solid 1px lightgray'
      }
    }
  };
});
var _default = useStyles;
exports.default = _default;
