// common-ui/lib/components/ValidatedTextField/ValidatedTextField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _TextField = _interopRequireDefault(require("../TextField"));
var _printVersion = _interopRequireDefault(require("../../utils/printVersion"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ValidatedTextField(props) {
  (0, _react.useEffect)(function () {
    return (0, _printVersion.default)();
  }, []);
  var _useState = (0, _react.useState)(props.value ? props.value : ''),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    content = _useState2[0],
    setContent = _useState2[1];
  var _useState3 = (0, _react.useState)(true),
    _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
    isValid = _useState4[0],
    setIsValid = _useState4[1];
  var _useState5 = (0, _react.useState)(''),
    _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
    errorMessage = _useState6[0],
    setErrorMessage = _useState6[1];
  (0, _react.useEffect)(function () {
    setContent(props.value);
    if (props.value && props.value.length || props.error) {
      checkValidations();
    }
  }, [props.value, props.error]);
  (0, _react.useEffect)(function () {
    if (props.onValidate) {
      props.onValidate(isValid);
    }
  }, [isValid]);
  var checkValidations = function checkValidations() {
    if (props.required && (!content || content.length === 0)) {
      setIsValid(false);
      setErrorMessage('שדה חובה');
      return;
    }
    if (content && props.maxLength && content.length > props.maxLength) {
      setIsValid(false);
      setErrorMessage("\u05DE\u05E1\u05E4\u05E8 \u05D4\u05EA\u05D5\u05D5\u05D9\u05DD \u05D4\u05DE\u05E7\u05E1\u05D9\u05DE\u05DC\u05D9 \u05D4\u05D5\u05D0 ".concat(props.maxLength));
      return;
    }
    if (content && props.minLength && content.length < props.minLength) {
      setIsValid(false);
      setErrorMessage("\u05DE\u05E1\u05E4\u05E8 \u05D4\u05EA\u05D5\u05D5\u05D9\u05DD \u05D4\u05DE\u05D9\u05E0\u05D9\u05DE\u05DC\u05D9 \u05D4\u05D5\u05D0 ".concat(props.minLength));
      return;
    }
    if (props.validationHandler(content)) {
      setIsValid(false);
      setErrorMessage(props.errorMessage);
      return;
    }
    setIsValid(true);
  };
  var onBlurHandler = function onBlurHandler(e) {
    checkValidations();
    if (props.onBlur) {
      props.onBlur(e);
    }
  };
  var onChangeHandler = function onChangeHandler(e) {
    setIsValid(true);
    setContent(e.target.value);
    if (props.onChange) {
      props.onChange(e);
    }
  };
  var extraProps = {
    type: !!props.isNumeric ? 'tel' : 'text',
    inputProps: {
      maxLength: props.maxLength,
      minLength: props.minLength
    }
  };
  return /*#__PURE__*/_react.default.createElement(_TextField.default, (0, _extends2.default)({}, extraProps, props, {
    value: content,
    error: !isValid,
    helperText: isValid ? null : errorMessage,
    onBlur: onBlurHandler,
    onChange: onChangeHandler
  }));
}
ValidatedTextField.propTypes = {
  /** Show numeric keyboard on mobile */
  isNumeric: _propTypes.default.bool,
  /** Maximum allowed characters */
  maxLength: _propTypes.default.number,
  /** Minimum allowed characters */
  minLength: _propTypes.default.number,
  /** Is field required */
  required: _propTypes.default.bool,
  /** Function that checks the input value, returns bool */
  validationHandler: _propTypes.default.func.isRequired,
  /** Error message if value is not valid */
  errorMessage: _propTypes.default.string.isRequired,
  /** Callback for validation check */
  onValidate: _propTypes.default.func,
  error: _propTypes.default.bool,
  value: _propTypes.default.string,
  onBlur: _propTypes.default.func,
  onChange: _propTypes.default.func
};
ValidatedTextField.defaultProps = {
  minLength: 0,
  maxLength: Number.MAX_SAFE_INTEGER,
  isNumeric: false,
  required: false,
  value: '',
  validationHandler: function validationHandler(value) {},
  errorMessage: 'ערך לא תקין',
  onValidate: function onValidate(isValid) {},
  error: false
};
var _default = ValidatedTextField;
exports.default = _default;
