// common-ui/lib/components/FileUpload/dragAndDrop/DragAndDrop.style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _colors = require("@material-ui/core/colors");

var styles = {
  dropZone: {
    fontSize: '18px',
    display: 'inline-block',
    verticalAlign: 'middle',
    width: '100%',
    cursor: 'pointer',
    '&.$disabled': {
      cursor: 'no-drop',
      '& $zoneAvatar': {
        background: '#ccc'
      },
      '& $zoneAvatarSvg': {
        color: 'whitesmoke'
      },
      '& $uploadZone': {
        backgroundColor: 'whitesmoke'
      },
      '& $uploadLink': {
        color: _colors.grey[800],
        cursor: 'not-allowed'
      }
    }
  },
  uploadZone: {
    textAlign: 'center',
    border: '1px dashed rgba(0,0,0,0.15)',
    borderRadius: '10px',
    padding: '20px',
    background: '#fff',
    position: 'relative'
  },
  uploadZoneDragging: {
    border: '3px dashed rgb(57,57,58)',
    background: 'rgb(240,240,241)'
  },
  zoneAvatar: {
    minWidth: 50,
    width: 50,
    height: 50,
    background: '#dfebf9',
    borderRadius: '10em',
    marginLeft: '16px',
    fontWeight: 900,
    display: 'inline-block'
  },
  zoneAvatarSvg: {
    marginTop: 5,
    color: '#006cb2'
  },
  zoneImage: {
    width: '100%',
    height: '100%',
    fill: '#003f80'
  },
  fileInput: {
    opacity: '0',
    border: 'none',
    borderRadius: '3px',
    background: 'grey',
    left: '0px',
    width: '100%',
    top: '0px',
    height: '100%',
    display: 'none'
  },
  uploadLink: {
    color: '#006cb2',
    textDecoration: 'none',
    fontWeight: 'bold' //or 700

  },
  zoneText: {
    color: _colors.grey[800],
    display: 'flex',
    flexDirection: 'column',
    marginTop: 10,
    textAlign: 'right',
    lineHeight: '17px',
    fontSize: 15
  },
  column: {
    display: 'flex',
    justifyContent: 'center'
  }
};
var _default = styles;
exports.default = _default;
