// harelkit/lib/components/Grid/Col.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Col = _interopRequireDefault(require("./Col.style"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _clsx = _interopRequireDefault(require("clsx"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["", ""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var StyledCol = (0, _styledComponents.default)('div')(_templateObject(), _Col.default);

var Col = function Col(props) {
  var children = props.children,
      xs = props.xs,
      sm = props.sm,
      md = props.md,
      lg = props.lg,
      xl = props.xl,
      xsOffset = props.xsOffset,
      smOffset = props.smOffset,
      mdOffset = props.mdOffset,
      lgOffset = props.lgOffset,
      xlOffset = props.xlOffset,
      className = props.className,
      vspacing = props.vspacing;
  var classNames = [];
  var classMap = {
    vspacing: 'vspacing',
    xs: 'col-xs',
    sm: 'col-sm',
    md: 'col-md',
    lg: 'col-lg',
    xl: 'col-xl',
    xsOffset: 'col-xs-offset',
    smOffset: 'col-sm-offset',
    mdOffset: 'col-md-offset',
    lgOffset: 'col-lg-offset',
    xlOffset: 'col-xl-offset'
  };

  var getClassNames = function getClassNames() {
    var list = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    var extraClasses = [];
    if (list.className) extraClasses.push(list.className);
    return Object.keys(list).filter(function (key) {
      return classMap[key];
    }).map(function (key) {
      return props[key] || props[key] === 0 ? "".concat(classMap[key], "-").concat(props[key]) : null;
    }).concat(extraClasses);
  };

  classNames.push(getClassNames({
    className: className,
    xs: xs,
    sm: sm,
    md: md,
    lg: lg,
    xl: xl,
    xsOffset: xsOffset,
    smOffset: smOffset,
    mdOffset: mdOffset,
    lgOffset: lgOffset,
    xlOffset: xlOffset,
    vspacing: vspacing
  }));
  return /*#__PURE__*/_react.default.createElement(StyledCol, {
    className: (0, _clsx.default)(classNames)
  }, children);
};

Col.propTypes = {
  /**
   * Defines the vertical spacing of current column
   */
  vspacing: _propTypes.default.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** 
   * Defines the number of columns component is going to use. 
   * It's applied for the xs breakpoint and wider screens if not overridden.
   */
  xs: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** 
   * Defines the number of columns component is going to use. 
   * It's applied for the sm breakpoint and wider screens if not overridden.
   */
  sm: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** 
   * Defines the number of columns component is going to use. 
   * It's applied for the md breakpoint and wider screens if not overridden.
   */
  md: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** 
   * Defines the number of columns component is going to use. 
   * It's applied for the lg breakpoint and wider screens if not overridden.
   */
  lg: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** 
   * Defines the number of columns component is going to use. 
   * It's applied for the xl breakpoint and wider screens if not overridden.
   */
  xl: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
  xsOffset: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
  smOffset: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
  mdOffset: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
  lgOffset: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
  xlOffset: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11])
};
Col.defaultProps = {
  xs: null,
  sm: null,
  md: null,
  lg: null,
  xl: null,
  xsOffset: null,
  smOffset: null,
  mdOffset: null,
  lgOffset: null,
  xlOffset: null
};
var _default = Col;
exports.default = _default;
