// harelkit/lib/components/Grid/Grid.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Grid = _interopRequireDefault(require("./Grid.style"));

var _Col = _interopRequireDefault(require("./Col"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _clsx = _interopRequireDefault(require("clsx"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["", ""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var StyledGrid = (0, _styledComponents.default)('div')(_templateObject(), _Grid.default);

var Grid = function Grid(props) {
  var children = props.children,
      verticalSpacing = props.verticalSpacing,
      reverse = props.reverse,
      justify = props.justify,
      alignItems = props.alignItems,
      className = props.className,
      rest = _objectWithoutProperties(props, ["children", "verticalSpacing", "reverse", "justify", "alignItems", "className"]);

  var classNames = [];
  if (className) classNames.push(className);
  if (reverse) classNames.push('reverse');
  if (verticalSpacing === 0) classNames.push("vspacing-0");
  if (verticalSpacing) classNames.push("vspacing-".concat(verticalSpacing));
  if (justify) classNames.push("justify-".concat(justify));
  if (alignItems) classNames.push("align-".concat(alignItems));
  return /*#__PURE__*/_react.default.createElement(StyledGrid, _extends({
    className: (0, _clsx.default)(classNames)
  }, rest), children);
};

Grid.Col = _Col.default;
Grid.propTypes = {
  /** 
   * Defines the space between the rows.
   */
  verticalSpacing: _propTypes.default.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),

  /**
   * reverse row order
   */
  reverse: _propTypes.default.bool,

  /** 
   * Defines the justify-content style property. 
   * It is applied for all screen sizes.
   */
  justify: _propTypes.default.oneOf(['start', 'center', 'end', 'between', 'around', 'evenly']),

  /** 
   * Defines the align-items style property. 
   * It's applied for all screen sizes. 
   */
  alignItems: _propTypes.default.oneOf(['start', 'center', 'end', 'stretch', 'baseline']),

  /**
   * Add css class to grid
   */
  className: _propTypes.default.string
};
Grid.defaultProps = {};
var _default = Grid;
exports.default = _default;
