// common-ui/lib/components/Drawer/Header.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _commonIcons = require("common-icons");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Header = _interopRequireDefault(require("./Header.style"));

var _clsx2 = _interopRequireDefault(require("clsx"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function Header(props) {
  var title = props.title,
      subtitle = props.subtitle,
      showClose = props.showClose,
      setClose = props.setClose;
  var classes = (0, _Header.default)();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx2.default)(classes.root, (0, _defineProperty2.default)({}, classes.hasClose, showClose))
  }, title && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.title
  }, title), subtitle && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.subtitle
  }, subtitle), showClose && /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.close,
    onClick: setClose,
    "data-hrl-bo": "close"
  }, /*#__PURE__*/_react.default.createElement(_commonIcons.IconX, null)));
}

Header.propTypes = {
  /**
   * Drawer title
   */
  title: _propTypes.default.node,

  /**
   * Drawer subtitle
   */
  subtitle: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.element]),
  showClose: _propTypes.default.bool,
  setClose: _propTypes.default.func
};
var _default = Header;
exports.default = _default;
