// common-ui/lib/components/AvatarInput/AvatarInput.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _core = require("@material-ui/core");
var useStyles = (0, _core.makeStyles)(function (theme) {
  var palette = theme.palette;
  return {
    root: {
      width: 118,
      height: 118,
      pointerEvents: function pointerEvents(props) {
        return props.disabled ? 'none' : 'auto';
      },
      opacity: function opacity(props) {
        return props.disabled ? 0.5 : 1;
      },
      filter: function filter(props) {
        return props.disabled && 'grayscale(100%)' || 'none';
      },
      position: 'relative',
      '& input ~ span': {
        transition: 'all 0.1s',
        opacity: 0
      },
      '&:hover input ~ div, & input:focus ~ div': {
        boxShadow: '0 0 0 6px rgba(228, 233, 242, 0.7)'
      },
      '& input:checked ~ div': {
        boxShadow: function boxShadow(props) {
          return '0 0 0 6px ' + (props.error ? palette.error.superLight : palette.secondary.superLight);
        }
      },
      '& input:checked ~ span': {
        opacity: 1
      }
    },
    container: {
      overflow: 'hidden',
      padding: theme.spacing(1.5),
      borderRadius: theme.spacing(10),
      border: function border(props) {
        return '1px dashed ' + (props.error ? palette.error.main : palette.secondary.main);
      },
      textAlign: 'center',
      cursor: 'pointer',
      width: '100%',
      height: '100%',
      transition: 'all 0.3s',
      userSelect: 'none',
      WebkitUserSelect: 'none',
      msUserSelect: 'none',
      MozUserSelect: 'none',
      '&:after': {
        content: '',
        position: 'absolute',
        display: 'none'
      }
    },
    checkbox: {
      position: 'absolute',
      width: '100%',
      height: '100%',
      opacity: 0,
      cursor: 'pointer',
      left: 0,
      margin: 0,
      top: 0
    },
    badge: {
      position: 'absolute',
      top: theme.spacing(0.3),
      right: theme.spacing(0.3),
      display: 'flex',
      width: 30,
      height: 30,
      backgroundColor: function backgroundColor(props) {
        return props.error ? palette.error.main : palette.success.main;
      },
      borderRadius: theme.spacing(10),
      color: '#fff',
      lineHeight: '32px',
      fontSize: 14,
      justifyContent: 'center',
      alignItems: 'center'
    }
  };
});
var _default = useStyles;
exports.default = _default;
