// common-ui/lib/components/FileUpload/progress/Progress.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var styles = {
  progressWrapper: {
    position: 'relative',
    height: '4px',
    display: 'block',
    width: '100%',
    backgroundColor: 'rgba(0,63,128,0.3)',
    borderRadius: '2px',
    overflow: 'hidden',
    margin: '6px 0'
  },
  progress: {
    position: 'absolute',
    top: '0',
    right: '0',
    bottom: '0',
    transition: 'width .3s linear',
    backgroundColor: 'rgba(0,63,128,0.7)',
    width: function width(props) {
      return props.step ? "".concat(props.step, "%") : 0;
    }
  },
  status: {
    position: 'absolute',
    visibility: 'hidden',
    top: '-9999px',
    left: '-9999px'
  }
};

var Progress = function Progress(props) {
  var accessibilityProps = {
    role: 'progressbar',
    tabIndex: -1,
    'aria-valuemin': 0,
    'aria-valuemax': 100,
    'aria-valuenow': props.step,
    'aria-describedby': "".concat(props.id, "-status")
  };

  var getStatusMessage = function getStatusMessage() {
    return props.hasError ? 'העלאה נכשלה' : props.step === 100 ? 'העלאה הסתיימה' : 'העלאה בתהליך';
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: props.classes.progressWrapper
  }, /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
    className: props.classes.progress
  }, accessibilityProps, props)), /*#__PURE__*/_react.default.createElement("span", {
    id: "".concat(props.id, "-status"),
    className: props.classes.status,
    hidden: true
  }, getStatusMessage()));
};

Progress.propTypes = {
  id: _propTypes.default.string,
  step: _propTypes.default.number,
  hasError: _propTypes.default.bool,
  classes: _propTypes.default.any
};
Progress.defaultProps = {
  step: 0
};

var _default = (0, _withStyles.default)(styles)(Progress);

exports.default = _default;
